<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>CITEseq-HIVE Seurat Cluster and ID cell types with Multimodal Data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/clipboard/clipboard.min.js"></script>
<script src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/quarto-html/quarto.js"></script>
<script src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/quarto-html/popper.min.js"></script>
<script src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/quarto-html/anchor.min.js"></script>
<link href="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">CITEseq-HIVE Seurat Cluster and ID cell types with Multimodal Data</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="description" class="level2">
<h2 class="anchored" data-anchor-id="description">Description</h2>
<p>This script takes the output of CITEseqHIVE_Seurat_FilterMergeQC.qmd as input. Each Assay (“RNA”, and “ADT”) will be processed individually first, and then integrated using Weighted Nearest Neighbors (WNN) for combined analysis. Analysis in this script includes normalization and scaling, clustering, and cell type identification.</p>
<section id="clear-console" class="level3">
<h3 class="anchored" data-anchor-id="clear-console">Clear console</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span>()</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-output-directory" class="level3">
<h3 class="anchored" data-anchor-id="set-output-directory">Set output directory</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>dir_save <span class="ot">&lt;-</span> <span class="st">"output/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-libraries" class="level3">
<h3 class="anchored" data-anchor-id="load-libraries">Load libraries</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratObject)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clustree)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(HGNChelper)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(multtest)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(metap)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(presto)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="rna" class="level2">
<h2 class="anchored" data-anchor-id="rna">RNA</h2>
<p>Begin by processing the “RNA” assay. In the below section, only RNA expression data will be considered when performing normalization, scaling, clustering, and cell identification steps. ADT data is ignored.</p>
<section id="load-the-filtered-merged-.rdata-object-and-examine-structure" class="level3">
<h3 class="anchored" data-anchor-id="load-the-filtered-merged-.rdata-object-and-examine-structure">Load the filtered, merged, .Rdata object and examine structure</h3>
<p>The Seurat_FilterMerge_obj.Rdata file generated by CITEseqHIVE_Seurat_FilterMergeQC.qmd serves as input for this pipeline. It should contain only one object, called “obj” which is the filtered and merged Seruat object for all SMARTseq2 plates.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"output/Seurat_FilterMerge_obj.Rdata"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>obj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
54869 features across 66972 samples within 3 assays 
Active assay: RNA (54608 features, 0 variable features)
 2 layers present: counts, data
 2 other assays present: ADT, HTO</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm Seruat object contains assays for RNA and ADT</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Assays</span>(obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RNA" "ADT" "HTO"</code></pre>
</div>
</div>
</details>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "RNA" "ADT" "HTO"</code></pre>
</div>
</div>
</section>
<section id="split-the-object-into-multiple-layers-for-integration-based-on-the-rna-assay" class="level3">
<h3 class="anchored" data-anchor-id="split-the-object-into-multiple-layers-for-integration-based-on-the-rna-assay">Split the object into multiple layers for integration based on the RNA Assay</h3>
<p>Integration is performed to account for batch effects. Therefore, the factor used to split the object into multiple layers should be the factor that provides information on which batch the samples came from. In this case, each batch has a unique RUN ID. The metadata column for that contains the RUN IDs is called “RUN” and should be used to integrate layers.</p>
<p>Because this Seurat object contains RNA, HTO, and ADT assays, we will perform integration separately beginning with the RNA assay.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">############## CONSIDER ADDING INTEGRATION FEATURES #######################</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>obj[[<span class="st">"RNA"</span>]] <span class="ot">&lt;-</span> <span class="fu">split</span>(obj[[<span class="st">"RNA"</span>]], <span class="at">f =</span> obj<span class="sc">$</span>RUN)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>obj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
54869 features across 66972 samples within 3 assays 
Active assay: RNA (54608 features, 0 variable features)
 6 layers present: counts.M72RUN001, counts.M72RUN002, counts.M72RUN005, data.M72RUN001, data.M72RUN002, data.M72RUN005
 2 other assays present: ADT, HTO</code></pre>
</div>
</div>
</details>
</section>
<section id="show-qc-metrics-for-the-first-5-cells-and-create-a-metadata-object" class="level3">
<h3 class="anchored" data-anchor-id="show-qc-metrics-for-the-first-5-cells-and-create-a-metadata-object">Show QC metrics for the first 5 cells and create a metadata object</h3>
<p>This is intended to give a snapshot look at some QC metrics and metadata features that can be used for downstream analysis.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(obj<span class="sc">@</span>meta.data, <span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">orig.ident</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nCount_RNA</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nFeature_RNA</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">SampleName</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">percent.mito</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">ExonReads</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Barcode</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">ExonvMapped</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">ExonvTotal</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">reads.mapped</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">reads.Total</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Complexity</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">IntronReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">IntronvTotal</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibTotReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibFiltReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibMappedReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibExonReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibIntronReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibPolyAReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Lib5PFReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Lib3PFReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibBadBaseReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibHTOReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">LibADTReads</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Ref</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">beeV</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">wdlV</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Dataset</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nGeneThresh</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nTranThresh</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">cellInput</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nCount_ADT</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nFeature_ADT</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">ADTUnmappedReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">ADTTotalReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nCount_HTO</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nFeature_HTO</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">HTOTotalReads</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">HTOUnmappedReads</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">MULTI_ID</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">RUN</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">PLATE</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">HIVE</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">PID</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">VISIT</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">STIM</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">group</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">sampleType</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">groupSamp</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">nFastq</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">TotalComplexity</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SeqSat</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SeqDepth</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">VISIT_STIM</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">VISIT_STIM_group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">_HIVE001_2</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: right;">2079</td>
<td style="text-align: right;">1204</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000183-D0-UNSTIM</td>
<td style="text-align: right;">0.0553151</td>
<td style="text-align: right;">8328</td>
<td style="text-align: left;">AAAAAAGCAAAT</td>
<td style="text-align: right;">0.7407935</td>
<td style="text-align: right;">0.7170656</td>
<td style="text-align: right;">11242</td>
<td style="text-align: right;">11614</td>
<td style="text-align: right;">4.005772</td>
<td style="text-align: right;">1351</td>
<td style="text-align: right;">0.1163251</td>
<td style="text-align: right;">242330169</td>
<td style="text-align: right;">240263917</td>
<td style="text-align: right;">230155552</td>
<td style="text-align: right;">110338716</td>
<td style="text-align: right;">46283567</td>
<td style="text-align: right;">216123</td>
<td style="text-align: right;">1810926</td>
<td style="text-align: right;">39203</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">57928525</td>
<td style="text-align: right;">126373229</td>
<td style="text-align: left;">20210603_GRCh38.104</td>
<td style="text-align: left;">v2.0rc1</td>
<td style="text-align: left;">v2.0</td>
<td style="text-align: left;">M72RUN001_M72CSH001</td>
<td style="text-align: right;">300</td>
<td style="text-align: right;">600</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1353</td>
<td style="text-align: right;">123</td>
<td style="text-align: right;">91</td>
<td style="text-align: right;">1444</td>
<td style="text-align: right;">1007</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">1230</td>
<td style="text-align: right;">223</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000183-D0-UNSTIM</td>
<td style="text-align: left;">M72RUN001</td>
<td style="text-align: left;">M72CSH001</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: left;">000183</td>
<td style="text-align: left;">D0</td>
<td style="text-align: left;">UNSTIM</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">5.586340</td>
<td style="text-align: right;">0.8209919</td>
<td style="text-align: right;">1.726744</td>
<td style="text-align: left;">D0 UNSTIM</td>
<td style="text-align: left;">D0 UNSTIM</td>
</tr>
<tr class="even">
<td style="text-align: left;">_HIVE001_7</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: right;">1437</td>
<td style="text-align: right;">932</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000106-D37-UNSTIM</td>
<td style="text-align: right;">0.0765484</td>
<td style="text-align: right;">4745</td>
<td style="text-align: left;">AAAAACTCGATA</td>
<td style="text-align: right;">0.7494867</td>
<td style="text-align: right;">0.7175261</td>
<td style="text-align: right;">6331</td>
<td style="text-align: right;">6613</td>
<td style="text-align: right;">3.302018</td>
<td style="text-align: right;">816</td>
<td style="text-align: right;">0.1233933</td>
<td style="text-align: right;">242330169</td>
<td style="text-align: right;">240263917</td>
<td style="text-align: right;">230155552</td>
<td style="text-align: right;">110338716</td>
<td style="text-align: right;">46283567</td>
<td style="text-align: right;">216123</td>
<td style="text-align: right;">1810926</td>
<td style="text-align: right;">39203</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">57928525</td>
<td style="text-align: right;">126373229</td>
<td style="text-align: left;">20210603_GRCh38.104</td>
<td style="text-align: left;">v2.0rc1</td>
<td style="text-align: left;">v2.0</td>
<td style="text-align: left;">M72RUN001_M72CSH001</td>
<td style="text-align: right;">300</td>
<td style="text-align: right;">600</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">2754</td>
<td style="text-align: right;">131</td>
<td style="text-align: right;">108</td>
<td style="text-align: right;">2862</td>
<td style="text-align: right;">2124</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">2404</td>
<td style="text-align: right;">280</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000106-D37-UNSTIM</td>
<td style="text-align: left;">M72RUN001</td>
<td style="text-align: left;">M72CSH001</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: left;">000106</td>
<td style="text-align: left;">D37</td>
<td style="text-align: left;">UNSTIM</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">4.601948</td>
<td style="text-align: right;">0.7827007</td>
<td style="text-align: right;">1.541845</td>
<td style="text-align: left;">D37 UNSTIM</td>
<td style="text-align: left;">D37 UNSTIM</td>
</tr>
<tr class="odd">
<td style="text-align: left;">_HIVE001_8</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: right;">1463</td>
<td style="text-align: right;">854</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000106-D37-UNSTIM</td>
<td style="text-align: right;">0.0956938</td>
<td style="text-align: right;">4484</td>
<td style="text-align: left;">AAAAAGATATTG</td>
<td style="text-align: right;">0.7022709</td>
<td style="text-align: right;">0.6819772</td>
<td style="text-align: right;">6385</td>
<td style="text-align: right;">6575</td>
<td style="text-align: right;">3.064935</td>
<td style="text-align: right;">776</td>
<td style="text-align: right;">0.1180228</td>
<td style="text-align: right;">242330169</td>
<td style="text-align: right;">240263917</td>
<td style="text-align: right;">230155552</td>
<td style="text-align: right;">110338716</td>
<td style="text-align: right;">46283567</td>
<td style="text-align: right;">216123</td>
<td style="text-align: right;">1810926</td>
<td style="text-align: right;">39203</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">57928525</td>
<td style="text-align: right;">126373229</td>
<td style="text-align: left;">20210603_GRCh38.104</td>
<td style="text-align: left;">v2.0rc1</td>
<td style="text-align: left;">v2.0</td>
<td style="text-align: left;">M72RUN001_M72CSH001</td>
<td style="text-align: right;">300</td>
<td style="text-align: right;">600</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1132</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">70</td>
<td style="text-align: right;">1202</td>
<td style="text-align: right;">1118</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">1273</td>
<td style="text-align: right;">155</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000106-D37-UNSTIM</td>
<td style="text-align: left;">M72RUN001</td>
<td style="text-align: left;">M72CSH001</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: left;">000106</td>
<td style="text-align: left;">D37</td>
<td style="text-align: left;">UNSTIM</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">4.494190</td>
<td style="text-align: right;">0.7774905</td>
<td style="text-align: right;">1.713115</td>
<td style="text-align: left;">D37 UNSTIM</td>
<td style="text-align: left;">D37 UNSTIM</td>
</tr>
<tr class="even">
<td style="text-align: left;">_HIVE001_12</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: right;">1437</td>
<td style="text-align: right;">884</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000183-D37-UNSTIM</td>
<td style="text-align: right;">0.0890745</td>
<td style="text-align: right;">4400</td>
<td style="text-align: left;">AAAAAGTGAATG</td>
<td style="text-align: right;">0.6986345</td>
<td style="text-align: right;">0.6247338</td>
<td style="text-align: right;">6298</td>
<td style="text-align: right;">7043</td>
<td style="text-align: right;">3.061935</td>
<td style="text-align: right;">519</td>
<td style="text-align: right;">0.0736902</td>
<td style="text-align: right;">242330169</td>
<td style="text-align: right;">240263917</td>
<td style="text-align: right;">230155552</td>
<td style="text-align: right;">110338716</td>
<td style="text-align: right;">46283567</td>
<td style="text-align: right;">216123</td>
<td style="text-align: right;">1810926</td>
<td style="text-align: right;">39203</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">57928525</td>
<td style="text-align: right;">126373229</td>
<td style="text-align: left;">20210603_GRCh38.104</td>
<td style="text-align: left;">v2.0rc1</td>
<td style="text-align: left;">v2.0</td>
<td style="text-align: left;">M72RUN001_M72CSH001</td>
<td style="text-align: right;">300</td>
<td style="text-align: right;">600</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">34623</td>
<td style="text-align: right;">152</td>
<td style="text-align: right;">1502</td>
<td style="text-align: right;">36125</td>
<td style="text-align: right;">2307</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">2603</td>
<td style="text-align: right;">296</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000183-D37-UNSTIM</td>
<td style="text-align: left;">M72RUN001</td>
<td style="text-align: left;">M72CSH001</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: left;">000183</td>
<td style="text-align: left;">D37</td>
<td style="text-align: left;">UNSTIM</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">4.901183</td>
<td style="text-align: right;">0.7959676</td>
<td style="text-align: right;">1.625566</td>
<td style="text-align: left;">D37 UNSTIM</td>
<td style="text-align: left;">D37 UNSTIM</td>
</tr>
<tr class="odd">
<td style="text-align: left;">_HIVE001_25</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: right;">1684</td>
<td style="text-align: right;">1035</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000183-D0-UNSTIM</td>
<td style="text-align: right;">0.0552257</td>
<td style="text-align: right;">5438</td>
<td style="text-align: left;">AAAACCTTCAAG</td>
<td style="text-align: right;">0.6464574</td>
<td style="text-align: right;">0.6176036</td>
<td style="text-align: right;">8412</td>
<td style="text-align: right;">8805</td>
<td style="text-align: right;">3.229216</td>
<td style="text-align: right;">1455</td>
<td style="text-align: right;">0.1652470</td>
<td style="text-align: right;">242330169</td>
<td style="text-align: right;">240263917</td>
<td style="text-align: right;">230155552</td>
<td style="text-align: right;">110338716</td>
<td style="text-align: right;">46283567</td>
<td style="text-align: right;">216123</td>
<td style="text-align: right;">1810926</td>
<td style="text-align: right;">39203</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">57928525</td>
<td style="text-align: right;">126373229</td>
<td style="text-align: left;">20210603_GRCh38.104</td>
<td style="text-align: left;">v2.0rc1</td>
<td style="text-align: left;">v2.0</td>
<td style="text-align: left;">M72RUN001_M72CSH001</td>
<td style="text-align: right;">300</td>
<td style="text-align: right;">600</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1797</td>
<td style="text-align: right;">125</td>
<td style="text-align: right;">48</td>
<td style="text-align: right;">1845</td>
<td style="text-align: right;">2395</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">2557</td>
<td style="text-align: right;">162</td>
<td style="text-align: left;">M72RUN001-M72CSH001-HIVE001-000183-D0-UNSTIM</td>
<td style="text-align: left;">M72RUN001</td>
<td style="text-align: left;">M72CSH001</td>
<td style="text-align: left;">HIVE001</td>
<td style="text-align: left;">000183</td>
<td style="text-align: left;">D0</td>
<td style="text-align: left;">UNSTIM</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">5.228622</td>
<td style="text-align: right;">0.8087450</td>
<td style="text-align: right;">1.627053</td>
<td style="text-align: left;">D0 UNSTIM</td>
<td style="text-align: left;">D0 UNSTIM</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> obj<span class="sc">@</span>meta.data</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details>
</section>
<section id="normalize-the-rna-data" class="level3">
<h3 class="anchored" data-anchor-id="normalize-the-rna-data">Normalize the RNA data</h3>
<p>As this dataset contains only an RNA assay we use standard Seurat NormalizeData function to perform log normalization with a scale factor of 10000.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Consider changing to SCTransform </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(obj, <span class="at">normalization.method =</span> <span class="st">"LogNormalize"</span>, <span class="at">scale.factor =</span> <span class="dv">10000</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details>
</section>
<section id="identify-variable-features" class="level3">
<h3 class="anchored" data-anchor-id="identify-variable-features">Identify variable features</h3>
<p>We select the top 2000 variable features for downstream analyses using the standard Seurat FindVariableFeatures function with selection method “vst”.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the top 2000 highly variable features</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(obj, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>) </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(obj), <span class="dv">10</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot variable features and label top 10 variable features</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(obj)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span><span class="fu">LabelPoints</span>(<span class="at">plot =</span> plot1, <span class="at">points =</span> top10, <span class="at">repel =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in scale_x_log10(): log-10 transformation introduced infinite values.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="scale-the-rna-data" class="level3">
<h3 class="anchored" data-anchor-id="scale-the-rna-data">Scale the RNA data</h3>
<p>Data is scaled using only variable features by default.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use defaults to scale only variable features </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(obj) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details>
</section>
<section id="perform-linear-dimensional-reduction" class="level3">
<h3 class="anchored" data-anchor-id="perform-linear-dimensional-reduction">Perform linear dimensional reduction</h3>
<p>Principal component analysis (PCA) is used for linear dimensional reduction of scaled data. PCA is run with default options. The plots below visualize dimensional reduction genes for principal components (PC) 1 and 2, display cell positioning in 2D, and a heatmap of genes associated with the top 1 and 15 PCs for all cells.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA for linear dimensional reduction on scaled data</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(obj, <span class="at">features =</span> <span class="cn">NULL</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  MALAT1, SYNE2, CCL5, SESN3, IL7R, CCND2, KLRD1, PRF1, GNLY, LRRN3 
       NCL, TIGIT, CD8A, NKG7, FGFBP2, CCR7, ITGA6, GZMH, MAF, MYC 
       ZBTB20, SYTL2, GZMA, GZMB, CEP78, CD8B, CEP290, RPAP2, NPM1, XAF1 
Negative:  CYP1B1, SLC7A11, MMP14, CREG1, CD68, CTSL, NCF2, KYNU, SOD2, C15ORF48 
       MARCKS, TNFAIP2, CXCL8, EMP1, IER3, LHFPL2, PILRA, CDKN1A, SERPINB2, TXN 
       ANPEP, TXNRD1, ANXA5, MGST1, PPIF, CSTB, RDX, IGSF6, TNFAIP6, LGALS3 
PC_ 2 
Positive:  HLA-DRA, MS4A1, CD74, HLA-DRB1, IGHM, HLA-DPA1, HLA-DQA1, HLA-DPB1, HLA-DQB1, BANK1 
       CD79A, HLA-DMB, SWAP70, CD79B, TCF4, MEF2C, WARS1, IGKC, HLA-DRB5, BCL11A 
       SPIB, RALGPS2, TNFRSF13B, CD40, IRF8, HLA-DOA, CD83, CYBB, CCDC50, ACP5 
Negative:  CCL5, IL36G, CCL3L1, CXCL3, IL7R, CXCL2, CCL20, SERPINB10, IL1A, CCL3 
       AQP9, CXCL1, EREG, SYNE2, ANXA1, GNLY, PRF1, KLRD1, SERPINB2, SERPINB7 
       IL6, IL1B, TNIP3, PID1, SRGN, TFPI2, TNFRSF1B, CXCL5, IL24, FGFBP2 
PC_ 3 
Positive:  CCL20, IL6, MS4A1, CCL3L1, TFPI2, IL36G, CXCL3, CXCL2, PALM2AKAP2, IL1A 
       AQP9, IGHM, SERPINB7, GNG11, CCL3, CXCL1, BANK1, CD79A, EBF1, CD79B 
       ATP2B1, IL24, IGKC, MEF2C, ENSG00000236453, SWAP70, SERPINB10, CCL3-AS1, NEDD4L, CXCL5 
Negative:  LYZ, IGSF6, CALHM6, GPNMB, PSTPIP2, SNX10, CCR1, LILRB3, CXCL10, RGL1 
       CXCL9, CCR5, ANKRD22, KCTD12, TLR4, CYP27A1, SECTM1, IL3RA, PLA2G7, CSF2RB 
       ALDH1A1, SLC31A2, EMP1, LILRB4, ATF3, APOL3, TYROBP, CTSB, RBM47, IDO1 
PC_ 4 
Positive:  PRF1, GNLY, KLRD1, CCL5, GZMB, NKG7, GZMH, FGFBP2, FCGR3A, PLEK 
       GZMA, C1ORF21, AKAP5, TRDC, ZEB2, SYTL2, TYROBP, LILRB1, TIGIT, SLAMF7 
       NCAM1, CTSC, CEP78, CADM1, HLA-DPB1, CMKLR1, TNFRSF1B, CD63, CCL4, METTL7A 
Negative:  IL7R, CCR7, TMEM123, LRRN3, MYC, NPM1, ITGA6, IL6ST, SESN3, IL6R 
       HSP90AB1, TSHZ2, TBC1D4, HSPD1, HBB, TNFAIP8, NUCB2, SCD, LIMS1, STAT1 
       TKT, HSPE1, SOCS3, CCR4, VIM, SNX9, CERS6, GBP1, TNFSF13B, AHR 
PC_ 5 
Positive:  FSCN1, LAMP3, CLIC2, LAD1, CCL22, PSD3, EBI3, UBD, PALLD, RASSF4 
       GABBR1, CRLF2, FSD1L, ZNF366, TFPI, CD86, TSPAN33, ALDH2, PVR, EHF 
       ROR1-AS1, ABTB2, GPR157, NR4A3, CKB, TVP23A, CD83, ADAM19, POGLUT1, ANKRD33B 
Negative:  CD14, EMP1, C5AR1, CCR1, GAPT, S100A9, MS4A1, IGHM, LACC1, MAFB 
       CCL2, CD79B, CYBB, ENSG00000250771, BANK1, FCAR, PLAUR, DUSP6, IGSF6, CD79A 
       DMXL2, GPNMB, PTAFR, TLR2, C3, RALGPS2, LILRB3, ITGAX, FAM111B, GRN </code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine and visualize PCA results a few different ways</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">VizDimLoadings</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"pca"</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">DimHeatmap</span>(obj, <span class="at">dims =</span> <span class="dv">1</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">DimHeatmap</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</details>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-13-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="examine-variance-explained-by-each-principal-component" class="level3">
<h3 class="anchored" data-anchor-id="examine-variance-explained-by-each-principal-component">Examine variance explained by each principal component</h3>
<p>Here we look at the variance in the dataset that is explained by each individual PC and at the cumulative variance explained by the PCs.</p>
<p>We also determine the dimensionality of the dataset using an Elbow plot (more efficient alternative to JackStraw procedure).</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">=</span> obj[[<span class="st">"pca"</span>]]</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the eigenvalues</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>evs <span class="ot">=</span> pca<span class="sc">@</span>stdev<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>total.var <span class="ot">=</span> pca<span class="sc">@</span>misc<span class="sc">$</span>total.variance</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>varExplained <span class="ot">=</span> evs<span class="sc">/</span>total.var</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>pca.data <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">PC=</span><span class="fu">factor</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(evs)),</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">percVar=</span>varExplained<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>pca.data<span class="sc">$</span>cumulVar <span class="ot">=</span> <span class="fu">cumsum</span>(pca.data<span class="sc">$</span>percVar)</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pca.data, <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   PC    percVar cumulVar
1   1 10.8118315 10.81183
2   2  2.1741837 12.98602
3   3  1.5604313 14.54645
4   4  1.0767560 15.62320
5   5  0.8969070 16.52011
6   6  0.7257520 17.24586
7   7  0.5929019 17.83876
8   8  0.5102662 18.34903
9   9  0.4751354 18.82416
10 10  0.3826380 19.20680
11 11  0.3537760 19.56058
12 12  0.3228157 19.88339
13 13  0.3020213 20.18542
14 14  0.3016430 20.48706
15 15  0.2858608 20.77292
16 16  0.2753251 21.04824
17 17  0.2604674 21.30871
18 18  0.2424085 21.55112
19 19  0.2337510 21.78487
20 20  0.2192903 22.00416</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>scPlot1 <span class="ot">&lt;-</span> pca.data[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,] <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC, <span class="at">y=</span>percVar)) <span class="sc">+</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">colour=</span><span class="st">"red"</span>, <span class="at">linetype=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Variance Explanation by PCA"</span>) <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>          <span class="fu">xlab</span>(<span class="st">"Principal Components"</span>) <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ylab</span>(<span class="st">"Percentage of Explained Variance"</span>) <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_bw</span>()</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>scPlot2 <span class="ot">&lt;-</span> pca.data[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,] <span class="sc">%&gt;%</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC, <span class="at">y=</span>cumulVar)) <span class="sc">+</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">'identity'</span>) <span class="sc">+</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">50</span>, <span class="at">colour=</span><span class="st">"red"</span>, <span class="at">linetype=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Cumulative Variance Explanation by PCA"</span>) <span class="sc">+</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>          <span class="fu">xlab</span>(<span class="st">"Principal Components"</span>) <span class="sc">+</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>          <span class="fu">ylab</span>(<span class="st">"Cumulative Percentage of Explained Variance"</span>) <span class="sc">+</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_bw</span>()</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>elbow <span class="ot">&lt;-</span> <span class="fu">ElbowPlot</span>(obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-16-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-16-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cluster-cells-based-on-rna" class="level3">
<h3 class="anchored" data-anchor-id="cluster-cells-based-on-rna">Cluster cells based on RNA</h3>
<p>Start with standard Seurat clustering method WITHOUT INTEGRATION. Use the bar and Elbow plots to determine minimum dimensions to use for clustering. Clustering is performed using UMAP for non-linear dimensional reduction. The number of dimensions is set to 30. PCA is used to find neighbors and clusters and for UMAP reduction.</p>
<p>The tabulated plots visualize the basic UMAP with seurat clusters for all cells and conditions as well as UMAPs grouped by metadata features and split by group.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(obj, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(obj, <span class="at">cluster.name =</span> <span class="st">"unintegrated_clusters"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 1809406

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8527
Number of communities: 13
Elapsed time: 20 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">reduction =</span> <span class="st">"pca"</span>, <span class="at">reduction.name =</span> <span class="st">"umap.unintegrated"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
This message will be shown once per session</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"RUN"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"PID"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>plot5 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>plot6 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT_STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" href="">UMAP group</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false" href="">UMAP RUN</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false" href="">UMAP STIM</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false" href="">UMAP PID</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-5" role="tab" aria-controls="tabset-1-5" aria-selected="false" href="">UMAP VISIT</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-6" role="tab" aria-controls="tabset-1-6" aria-selected="false" href="">UMAP VISIT &amp; STIM</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-5-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-6-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="inspect-cluster-proportions-on-unintegrated-clusters" class="level3">
<h3 class="anchored" data-anchor-id="inspect-cluster-proportions-on-unintegrated-clusters">Inspect cluster proportions on unintegrated clusters</h3>
<p>Next we examine the proportion of each cluster that is comprised of cells with key metadata features: RUN, group, STIM, PID, and VISIT. Proportions are shown alongside total cell number per cluster.</p>
<details>
<summary>
Code
</summary>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>plot_integrated_clusters <span class="ot">=</span> <span class="cf">function</span> (srat, batchcolumn) {</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="do">## take an integrated Seurat object, plot distributions over orig.ident</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  count_table <span class="ot">&lt;-</span> <span class="fu">table</span>(srat<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters, srat<span class="sc">@</span>meta.data[[batchcolumn]])</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  count_mtx   <span class="ot">&lt;-</span> <span class="fu">as.data.frame.matrix</span>(count_table)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  count_mtx<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="fu">rownames</span>(count_mtx)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  melt_mtx    <span class="ot">&lt;-</span> <span class="fu">melt</span>(count_mtx)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  melt_mtx<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(melt_mtx<span class="sc">$</span>cluster)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  cluster_size   <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(value <span class="sc">~</span> cluster, <span class="at">data =</span> melt_mtx, <span class="at">FUN =</span> sum)</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  sorted_labels <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">sort</span>(<span class="fu">as.integer</span>(<span class="fu">levels</span>(cluster_size<span class="sc">$</span>cluster)),<span class="at">decreasing =</span> T))</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  cluster_size<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="fu">factor</span>(cluster_size<span class="sc">$</span>cluster,<span class="at">levels =</span> sorted_labels)</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  melt_mtx<span class="sc">$</span>cluster <span class="ot">&lt;-</span> <span class="fu">factor</span>(melt_mtx<span class="sc">$</span>cluster,<span class="at">levels =</span> sorted_labels)</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(melt_mtx)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"dataset"</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(cluster_size, <span class="fu">aes</span>(<span class="at">y=</span> cluster,<span class="at">x =</span> value)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">"dodge"</span>, <span class="at">stat=</span><span class="st">"identity"</span>,<span class="at">fill =</span> <span class="st">"grey60"</span>) <span class="sc">+</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">scale_x_log10</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Cells per cluster, log10 scale"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">""</span>)</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(melt_mtx,<span class="fu">aes</span>(<span class="at">x=</span>cluster,<span class="at">y=</span>value,<span class="at">fill=</span>dataset)) <span class="sc">+</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">"fill"</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_fill_brewer(palette = "Set2") +</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Fraction of cells in each dataset"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Cluster number"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"top"</span>)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>  p2 <span class="sc">+</span> p1 <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>))</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'group'</span>)</span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'RUN'</span>)</span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'STIM'</span>)</span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'PID'</span>)</span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>plot5 <span class="ot">&lt;-</span> <span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'VISIT'</span>)</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a>plot6 <span class="ot">&lt;-</span> <span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'VISIT_STIM'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</details>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true" href="">Proportions group</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false" href="">Proportions RUN</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-3" role="tab" aria-controls="tabset-2-3" aria-selected="false" href="">Proportions STIM</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-4" role="tab" aria-controls="tabset-2-4" aria-selected="false" href="">Proportions PID</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-5" role="tab" aria-controls="tabset-2-5" aria-selected="false" href="">Proportions VISIT</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-6" role="tab" aria-controls="tabset-2-6" aria-selected="false" href="">Proportions VISIT &amp; STIM</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-2-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-3-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-2-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-4-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-2-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-5-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-2-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-6-tab">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="perform-integration-of-rna-assay" class="level3">
<h3 class="anchored" data-anchor-id="perform-integration-of-rna-assay">Perform integration of RNA Assay</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">IntegrateLayers</span>(<span class="at">object =</span> obj, </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> CCAIntegration, </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">orig.reduction =</span> <span class="st">"pca"</span>, </span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">new.reduction =</span> <span class="st">"integrated.cca"</span>, </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>obj[[<span class="st">"RNA"</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(obj[[<span class="st">"RNA"</span>]])</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(obj, <span class="at">reduction =</span> <span class="st">"integrated.cca"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>) </span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(obj, <span class="at">resolution =</span> <span class="fl">1.2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2646902

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8388
Number of communities: 23
Elapsed time: 29 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">reduction =</span> <span class="st">"integrated.cca"</span>) <span class="co"># Change back to 1:12</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"seurat_clusters"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"PID"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-32-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-32-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT_STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-32-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="inspect-run-metrics-on-integrated-clusters" class="level3">
<h3 class="anchored" data-anchor-id="inspect-run-metrics-on-integrated-clusters">Inspect RUN metrics on integrated clusters</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'RUN'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in melt.default(count_mtx): The melt generic in data.table has been
passed a data.frame and will attempt to redirect to the relevant reshape2
method; please note that reshape2 is superseded and is no longer actively
developed, and this redirection is now deprecated. To continue using melt
methods from reshape2 while both libraries are attached, e.g. melt.list, you
can prepend the namespace, i.e. reshape2::melt(count_mtx). In the next version,
this warning will become an error.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using cluster as id variables</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'group'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in melt.default(count_mtx): The melt generic in data.table has been
passed a data.frame and will attempt to redirect to the relevant reshape2
method; please note that reshape2 is superseded and is no longer actively
developed, and this redirection is now deprecated. To continue using melt
methods from reshape2 while both libraries are attached, e.g. melt.list, you
can prepend the namespace, i.e. reshape2::melt(count_mtx). In the next version,
this warning will become an error.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using cluster as id variables</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-33-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'STIM'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in melt.default(count_mtx): The melt generic in data.table has been
passed a data.frame and will attempt to redirect to the relevant reshape2
method; please note that reshape2 is superseded and is no longer actively
developed, and this redirection is now deprecated. To continue using melt
methods from reshape2 while both libraries are attached, e.g. melt.list, you
can prepend the namespace, i.e. reshape2::melt(count_mtx). In the next version,
this warning will become an error.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using cluster as id variables</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-33-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'PID'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in melt.default(count_mtx): The melt generic in data.table has been
passed a data.frame and will attempt to redirect to the relevant reshape2
method; please note that reshape2 is superseded and is no longer actively
developed, and this redirection is now deprecated. To continue using melt
methods from reshape2 while both libraries are attached, e.g. melt.list, you
can prepend the namespace, i.e. reshape2::melt(count_mtx). In the next version,
this warning will become an error.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using cluster as id variables</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-33-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'VISIT'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in melt.default(count_mtx): The melt generic in data.table has been
passed a data.frame and will attempt to redirect to the relevant reshape2
method; please note that reshape2 is superseded and is no longer actively
developed, and this redirection is now deprecated. To continue using melt
methods from reshape2 while both libraries are attached, e.g. melt.list, you
can prepend the namespace, i.e. reshape2::melt(count_mtx). In the next version,
this warning will become an error.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using cluster as id variables</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-33-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_integrated_clusters</span>(obj, <span class="st">'VISIT_STIM'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in melt.default(count_mtx): The melt generic in data.table has been
passed a data.frame and will attempt to redirect to the relevant reshape2
method; please note that reshape2 is superseded and is no longer actively
developed, and this redirection is now deprecated. To continue using melt
methods from reshape2 while both libraries are attached, e.g. melt.list, you
can prepend the namespace, i.e. reshape2::melt(count_mtx). In the next version,
this warning will become an error.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using cluster as id variables</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-33-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="determine-more-accurate-cluster-resolution-using-clustree" class="level3">
<h3 class="anchored" data-anchor-id="determine-more-accurate-cluster-resolution-using-clustree">Determine more accurate cluster resolution using clustree</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>resolution.range <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span><span class="fl">0.2</span>, <span class="at">to =</span> <span class="dv">2</span>, <span class="at">by =</span> <span class="fl">0.2</span>)</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> Seurat<span class="sc">::</span><span class="fu">FindClusters</span>(<span class="at">object =</span> obj, <span class="at">resolution =</span> resolution.range)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>clustree <span class="ot">&lt;-</span> <span class="fu">clustree</span>(obj, <span class="at">prefix =</span> <span class="st">"RNA_snn_res."</span>)</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"clustree.all.pdf"</span>))</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>clustree_plot <span class="ot">&lt;-</span> clustree</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(clustree_plot)</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>clustree_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>umap.<span class="fl">0.4</span> <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"RNA_snn_res.0.4"</span>)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>umap.<span class="fl">0.6</span> <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"RNA_snn_res.0.6"</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>umap.<span class="fl">0.8</span> <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"RNA_snn_res.0.8"</span>)</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>umap<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"RNA_snn_res.1"</span>)</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>umap.<span class="fl">1.2</span> <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"RNA_snn_res.1.2"</span>)</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>snn_all <span class="ot">&lt;-</span> (umap.<span class="fl">0.4</span> <span class="sc">|</span> umap.<span class="fl">0.6</span>) <span class="sc">/</span> (umap.<span class="fl">0.8</span> <span class="sc">|</span> umap<span class="fl">.1</span>) <span class="sc">/</span> umap.<span class="fl">1.2</span></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_snn_compare.pdf"</span>), <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">20</span>)</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>snn_all</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>snn_all</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(<span class="at">object =</span> obj) <span class="ot">&lt;-</span> <span class="st">"RNA_snn_res.1.2"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualize-clusters" class="level3">
<h3 class="anchored" data-anchor-id="visualize-clusters">Visualize clusters</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot UMAP</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_res.clustree.pdf"</span>))</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj,</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">6</span>)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot1)</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>plot1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot Feature plot to ensure clustering is not driven by QC artifacts</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_res.clustree_QC.pdf"</span>))</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>QC_feature <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nCount_RNA"</span>, <span class="st">"nFeature_RNA"</span>, <span class="st">"percent.mito"</span>, <span class="st">"Complexity"</span>))</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(QC_feature)</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>QC_feature</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-35-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allHIVES_res.clustree.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">25</span>)</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj,</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"HIVE"</span>,</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T)</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 12 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 12 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allPID_res.clustree.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">25</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj,</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"PID"</span>,</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T)</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 6 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>plot3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 6 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-36-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allSTIM_res.clustree.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj,</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"STIM"</span>,</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>plot4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-36-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allVISIT_res.clustree.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>plot5 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj,</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"VISIT"</span>,</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot5)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>plot5</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-36-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allConditions_res.clustree.pdf"</span>), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>plot6 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj,</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"STIM"</span>),</span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"VISIT"</span>,</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot6)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>plot6</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-36-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allConditionsAlternative_res.clustree.pdf"</span>), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>plot7 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj,</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"STIM"</span>),</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"group"</span>,</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plot7)</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>plot7</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-36-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="determine-marker-expression-by-cluster" class="level3">
<h3 class="anchored" data-anchor-id="determine-marker-expression-by-cluster">Determine marker expression by cluster</h3>
<p>Find markers for every cell cluster compared to all remaining cells, report only the positive ones.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>obj.markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(obj, <span class="at">only.pos =</span> <span class="cn">TRUE</span>, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>)</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>obj.markers <span class="sc">%&gt;%</span></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(obj.markers, <span class="at">file =</span> <span class="fu">file.path</span>(dir_save, <span class="st">"all_markers_snn.clustree.csv"</span>), <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="identify-markers-differentially-expressed-between-specific-clusters-of-interest-and-plot-differentially-expressed-features-as-relative-expression-and-as-counts." class="level3">
<h3 class="anchored" data-anchor-id="identify-markers-differentially-expressed-between-specific-clusters-of-interest-and-plot-differentially-expressed-features-as-relative-expression-and-as-counts.">Identify markers differentially expressed between specific clusters of interest and plot differentially expressed features as relative expression and as counts.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list of all cluster identities in obj</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>cluster.ids <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">Idents</span>(obj))</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store cluster markers</span></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>cluster.markers <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each Seurat cluster in obj</span></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cluster.id <span class="cf">in</span> cluster.ids) {</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find markers for the current cluster</span></span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>  cluster.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(obj, <span class="at">ident.1 =</span> cluster.id)</span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Assign a dynamic object name based on the cluster ID</span></span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>  object.name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster.markers."</span>, cluster.id)</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Export cluster markers to a CSV file</span></span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>  csv_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(object.name, <span class="st">".csv"</span>)</span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(cluster.markers, <span class="at">file =</span> <span class="fu">file.path</span>(dir_save, <span class="fu">paste0</span>(csv_file)), <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot expression of top 10 differentially expressed genes for the current cluster</span></span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>   vln_plot <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">features =</span> <span class="fu">rownames</span>(cluster.markers)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a>  vln_plot_relexp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_VlnPlotRelExp_Top10_"</span>, object.name, <span class="st">".pdf"</span>)</span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, vln_plot_relexp), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb82-28"><a href="#cb82-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(vln_plot)</span>
<span id="cb82-29"><a href="#cb82-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb82-30"><a href="#cb82-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb82-31"><a href="#cb82-31" aria-hidden="true" tabindex="-1"></a>  vln_plot <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb82-32"><a href="#cb82-32" aria-hidden="true" tabindex="-1"></a>          <span class="at">features =</span> <span class="fu">rownames</span>(cluster.markers)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb82-33"><a href="#cb82-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">slot =</span> <span class="st">"counts"</span>, </span>
<span id="cb82-34"><a href="#cb82-34" aria-hidden="true" tabindex="-1"></a>          <span class="at">log =</span> <span class="cn">TRUE</span>, </span>
<span id="cb82-35"><a href="#cb82-35" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb82-36"><a href="#cb82-36" aria-hidden="true" tabindex="-1"></a>          <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb82-37"><a href="#cb82-37" aria-hidden="true" tabindex="-1"></a>  vln_plot_counts <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_VlnPlotCounts_Top10_"</span>, object.name, <span class="st">".pdf"</span>)</span>
<span id="cb82-38"><a href="#cb82-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, vln_plot_counts), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb82-39"><a href="#cb82-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(vln_plot)</span>
<span id="cb82-40"><a href="#cb82-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb82-41"><a href="#cb82-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb82-42"><a href="#cb82-42" aria-hidden="true" tabindex="-1"></a>  feature_plot <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(obj,</span>
<span id="cb82-43"><a href="#cb82-43" aria-hidden="true" tabindex="-1"></a>            <span class="at">features =</span> <span class="fu">rownames</span>(cluster.markers)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], </span>
<span id="cb82-44"><a href="#cb82-44" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb82-45"><a href="#cb82-45" aria-hidden="true" tabindex="-1"></a>            <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb82-46"><a href="#cb82-46" aria-hidden="true" tabindex="-1"></a>            <span class="at">split.by =</span> <span class="st">"group"</span>,</span>
<span id="cb82-47"><a href="#cb82-47" aria-hidden="true" tabindex="-1"></a>            <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb82-48"><a href="#cb82-48" aria-hidden="true" tabindex="-1"></a>  feature_plots <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_FeaturePlot_Top10_"</span>, object.name, <span class="st">".pdf"</span>)</span>
<span id="cb82-49"><a href="#cb82-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, feature_plots), <span class="at">height =</span> <span class="dv">24</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb82-50"><a href="#cb82-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(feature_plot)</span>
<span id="cb82-51"><a href="#cb82-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb82-52"><a href="#cb82-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb82-53"><a href="#cb82-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb82-54"><a href="#cb82-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-55"><a href="#cb82-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot a heatmap to look at the top 10 differentially expressed genes across all clusters</span></span>
<span id="cb82-56"><a href="#cb82-56" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"DEG_Heatmap_AllClusters_Top10.pdf"</span>), <span class="at">height =</span> <span class="dv">30</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb82-57"><a href="#cb82-57" aria-hidden="true" tabindex="-1"></a>obj.markers <span class="sc">%&gt;%</span></span>
<span id="cb82-58"><a href="#cb82-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb82-59"><a href="#cb82-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">10</span>, <span class="at">wt =</span> avg_log2FC) <span class="ot">-&gt;</span> top10</span>
<span id="cb82-60"><a href="#cb82-60" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(obj, <span class="at">features =</span> top10<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb82-61"><a href="#cb82-61" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb82-62"><a href="#cb82-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-63"><a href="#cb82-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all markers distinguishing an interesting cluster from main PBMC clusters</span></span>
<span id="cb82-64"><a href="#cb82-64" aria-hidden="true" tabindex="-1"></a>InterestClustervOthers.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(obj, <span class="at">ident.1 =</span> <span class="dv">16</span>, <span class="at">ident.2 =</span> <span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">15</span>, <span class="dv">17</span><span class="sc">:</span><span class="dv">21</span>)) <span class="co">#Automate this line</span></span>
<span id="cb82-65"><a href="#cb82-65" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(InterestClustervOthers.markers, <span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       p_val avg_log2FC pct.1 pct.2 p_val_adj
IGSF6      0   4.309906 0.904 0.060         0
CD68       0   3.629429 0.929 0.112         0
EMP1       0   4.177656 0.873 0.074         0
IFI30      0   3.833942 0.987 0.191         0
NCF2       0   3.457172 0.882 0.094         0
LYZ        0   4.228209 0.868 0.082         0
LGALS3     0   3.334007 0.957 0.174         0
SNX10      0   4.290873 0.854 0.081         0
CYP1B1     0   3.356470 0.944 0.174         0
CTSB       0   3.772473 0.945 0.183         0</code></pre>
</div>
</div>
</section>
<section id="visualize-expression-of-sorting-panel-markers-on-clusters." class="level3">
<h3 class="anchored" data-anchor-id="visualize-expression-of-sorting-panel-markers-on-clusters.">Visualize expression of sorting panel markers on clusters.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot expression of the sorting panel markers </span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a> vln_plot <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"CD3G"</span>, <span class="st">"CD4"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD69"</span>, <span class="st">"CD40LG"</span>, <span class="st">"TNFRSF9"</span>),</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>        <span class="co">#pt.size = 0.1,</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">add.noise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>vln_plot_relexp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_VlnPlotRelExp_SortingPanel.pdf"</span>)</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, vln_plot_relexp), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(vln_plot)</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>vln_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>vln_plot <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"CD3G"</span>, <span class="st">"CD4"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD69"</span>, <span class="st">"CD40LG"</span>, <span class="st">"TNFRSF9"</span>),</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">slot =</span> <span class="st">"counts"</span>, </span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">log =</span> <span class="cn">TRUE</span>, </span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>        <span class="co">#pt.size = 0.1,</span></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">add.noise =</span> <span class="cn">FALSE</span>)</span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>vln_plot_counts <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_VlnPlotCounts_SortingPanel.pdf"</span>)</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, vln_plot_counts), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(vln_plot)</span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>vln_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-39-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>feature_plot <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(obj,</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"CD3G"</span>, <span class="st">"CD4"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD69"</span>, <span class="st">"CD40LG"</span>, <span class="st">"TNFRSF9"</span>), </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">split.by =</span> <span class="st">"group"</span>,</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>feature_plots <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_FeaturePlot_SortingPanel.pdf"</span>)</span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, feature_plots), <span class="at">height =</span> <span class="dv">24</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(feature_plot)</span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>feature_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-39-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="assign-cell-types" class="level3">
<h3 class="anchored" data-anchor-id="assign-cell-types">Assign cell types</h3>
<p>Predict cell types based on expression using sc-Type</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene set preparation function </span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R"</span>)</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load cell type annotation function</span></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Prepare gene sets from input cell marker file. This uses the default of in-built cell marker DB.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DB file</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="co">#db_ &lt;- "https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx";</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>db_ <span class="ot">&lt;-</span> <span class="st">"data/ScTypeDB_full_MLedit.xlsx"</span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>tissue <span class="ot">&lt;-</span> <span class="st">"Immune system"</span> <span class="co"># e.g. Immune system, Pancreas, Liver, Eye, Kidney, Brain, Lung, Adrenal, Heart, Intestine, Muscle, Placenta, Spleen, Stomach, Thymus </span></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare gene sets</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>gs_list <span class="ot">&lt;-</span> <span class="fu">gene_sets_prepare</span>(db_, tissue)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Run sc-Type.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="do">### CHANGE Idents(obj) to obj@meta.data$seurat_clusters</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="co"># check Seurat object version (scRNA-seq matrix extracted differently in Seurat v4/v5)</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>seurat_package_v5 <span class="ot">&lt;-</span> <span class="fu">isFALSE</span>(<span class="st">'counts'</span> <span class="sc">%in%</span> <span class="fu">names</span>(<span class="fu">attributes</span>(obj[[<span class="st">"RNA"</span>]])));</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">sprintf</span>(<span class="st">"Seurat object %s is used"</span>, <span class="fu">ifelse</span>(seurat_package_v5, <span class="st">"v5"</span>, <span class="st">"v4"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Seurat object v5 is used"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extract scaled scRNA-seq matrix</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>scRNAseqData_scaled <span class="ot">&lt;-</span> <span class="cf">if</span> (seurat_package_v5) <span class="fu">as.matrix</span>(obj[[<span class="st">"RNA"</span>]]<span class="sc">$</span>scale.data) <span class="cf">else</span> <span class="fu">as.matrix</span>(obj[[<span class="st">"RNA"</span>]]<span class="sc">@</span>scale.data)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="co"># run ScType</span></span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>es.max <span class="ot">&lt;-</span> <span class="fu">sctype_score</span>(<span class="at">scRNAseqData =</span> scRNAseqData_scaled, <span class="at">scaled =</span> <span class="cn">TRUE</span>, <span class="at">gs =</span> gs_list<span class="sc">$</span>gs_positive, <span class="at">gs2 =</span> gs_list<span class="sc">$</span>gs_negative)</span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a><span class="co"># merge by cluster</span></span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a>cL_resutls <span class="ot">&lt;-</span> <span class="fu">do.call</span>(<span class="st">"rbind"</span>, <span class="fu">lapply</span>(<span class="fu">unique</span>(<span class="fu">Idents</span>(obj)), <span class="cf">function</span>(cl){</span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a>  es.max.cl <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">rowSums</span>(es.max[ ,<span class="fu">rownames</span>(obj<span class="sc">@</span>meta.data[<span class="fu">Idents</span>(obj)<span class="sc">==</span>cl, ])]), <span class="at">decreasing =</span> <span class="sc">!</span><span class="dv">0</span>)</span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="fu">data.frame</span>(<span class="at">cluster =</span> cl, <span class="at">type =</span> <span class="fu">names</span>(es.max.cl), <span class="at">scores =</span> es.max.cl, <span class="at">ncells =</span> <span class="fu">sum</span>(<span class="fu">Idents</span>(obj)<span class="sc">==</span>cl)), <span class="dv">10</span>)</span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a>sctype_scores <span class="ot">&lt;-</span> cL_resutls <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">1</span>, <span class="at">wt =</span> scores)  </span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-14"><a href="#cb91-14" aria-hidden="true" tabindex="-1"></a><span class="co"># set low-confident (low ScType score) clusters to "unknown"</span></span>
<span id="cb91-15"><a href="#cb91-15" aria-hidden="true" tabindex="-1"></a>sctype_scores<span class="sc">$</span>type[<span class="fu">as.numeric</span>(<span class="fu">as.character</span>(sctype_scores<span class="sc">$</span>scores)) <span class="sc">&lt;</span> sctype_scores<span class="sc">$</span>ncells<span class="sc">/</span><span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">"Unknown"</span></span>
<span id="cb91-16"><a href="#cb91-16" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(sctype_scores)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 25 × 4
# Groups:   cluster [22]
   cluster type                        scores ncells
   &lt;fct&gt;   &lt;chr&gt;                        &lt;dbl&gt;  &lt;int&gt;
 1 4       Naive CD4+ T cells           2605.   4688
 2 1       Effector CD8+ T cells        7486.   5474
 3 3       Naive CD4+ T cells           2798.   5058
 4 15      Naive B cells                5466.   1952
 5 15      Memory B cells               5466.   1952
 6 9       Natural killer  cells       11694.   3086
 7 14      Myeloid Dendritic cells      3563.   2000
 8 7       CD4+ NKT-like cells          2599.   3877
 9 13      Natural killer  cells        6032.   2388
10 11      ISG expressing immune cells  1918.   2983
# ℹ 15 more rows</code></pre>
</div>
</div>
<p>Overlay cell type IDs on UMAP.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="do">### CHANGE Idents(obj) to obj@meta.data$seurat_clusters</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_classification <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> <span class="fu">unique</span>(sctype_scores<span class="sc">$</span>cluster)){</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>  cl_type <span class="ot">=</span> sctype_scores[sctype_scores<span class="sc">$</span>cluster<span class="sc">==</span>j,]; </span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>  obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_classification[<span class="fu">Idents</span>(obj) <span class="sc">==</span> j] <span class="ot">=</span> <span class="fu">as.character</span>(cl_type<span class="sc">$</span>type[<span class="dv">1</span>])</span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_annotated.pdf"</span>), <span class="at">height =</span> <span class="dv">7</span>, <span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>umap_annotated <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">group.by =</span> <span class="st">'sctype_classification'</span>)</span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(umap_annotated)</span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a>umap_annotated</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_annotated_splitgroup.pdf"</span>), <span class="at">height =</span> <span class="dv">7</span>, <span class="at">width =</span> <span class="dv">15</span>)</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>umap_annotated_group <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">split.by =</span> <span class="st">"group"</span>, <span class="at">group.by =</span> <span class="st">'sctype_classification'</span>)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(umap_annotated_group)</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>umap_annotated_group</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="create-proportion-plots-for-clusters" class="level3">
<h3 class="anchored" data-anchor-id="create-proportion-plots-for-clusters">Create proportion plots for clusters</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="do">########### DOWNSAMPLE TO SAME NUMBER OF CELLS PER VISIT PER GROUP</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> obj<span class="sc">@</span>meta.data</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>cluster_names <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(RNA_snn_res.<span class="fl">0.6</span>, sctype_classification) <span class="sc">%&gt;%</span> </span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sctype_classification =</span> <span class="fu">as.character</span>(sctype_classification))</span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(cluster_names) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>cluster_names <span class="ot">&lt;-</span> cluster_names <span class="sc">%&gt;%</span> </span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a><span class="co">#sctype</span></span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the total number of cells per sample</span></span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>total_cells_per_sample <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(VISIT, PID, sctype_classification) <span class="sc">%&gt;%</span></span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_cells =</span> <span class="fu">n</span>())</span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the number of cells per cluster per sample</span></span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>cells_per_cluster_sample <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(STIM, VISIT, PID, sctype_classification) <span class="sc">%&gt;%</span></span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">cluster_cells =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the proportion of cells in each cluster for each sample</span></span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>proportions <span class="ot">&lt;-</span> cells_per_cluster_sample <span class="sc">%&gt;%</span></span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(total_cells_per_sample, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"PID"</span>, <span class="st">"sctype_classification"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">proportion =</span> cluster_cells <span class="sc">/</span> total_cells) <span class="sc">%&gt;%</span></span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(STIM, VISIT, PID, sctype_classification, proportion)</span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot the data to create a matrix</span></span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a>proportion_df <span class="ot">&lt;-</span> proportions <span class="sc">%&gt;%</span></span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> sctype_classification, <span class="at">values_from =</span> proportion)</span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> proportion_df <span class="sc">%&gt;%</span> </span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">21</span>, <span class="at">names_to =</span> <span class="st">"sctype_classification"</span>, <span class="at">values_to =</span> <span class="st">"Prop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sctype_classification =</span> <span class="fu">as.character</span>(sctype_classification)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(cluster_names, <span class="at">by =</span> <span class="st">"sctype_classification"</span>) </span>
<span id="cb95-40"><a href="#cb95-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-41"><a href="#cb95-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot proportions</span></span>
<span id="cb95-42"><a href="#cb95-42" aria-hidden="true" tabindex="-1"></a>p_bar_cell <span class="ot">&lt;-</span> test <span class="sc">%&gt;%</span> </span>
<span id="cb95-43"><a href="#cb95-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> sctype_classification, <span class="at">y =</span> Prop, <span class="at">fill =</span> STIM)) <span class="sc">+</span></span>
<span id="cb95-44"><a href="#cb95-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>),</span>
<span id="cb95-45"><a href="#cb95-45" aria-hidden="true" tabindex="-1"></a>           <span class="at">stat =</span> <span class="st">"summary"</span>,</span>
<span id="cb95-46"><a href="#cb95-46" aria-hidden="true" tabindex="-1"></a>           <span class="at">fun =</span> <span class="st">"median"</span>,</span>
<span id="cb95-47"><a href="#cb95-47" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb95-48"><a href="#cb95-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(PID <span class="sc">~</span> VISIT) <span class="sc">+</span>  </span>
<span id="cb95-49"><a href="#cb95-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Proportion of cells"</span>) <span class="sc">+</span></span>
<span id="cb95-50"><a href="#cb95-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb95-51"><a href="#cb95-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">alpha =</span> <span class="st">"none"</span>,</span>
<span id="cb95-52"><a href="#cb95-52" aria-hidden="true" tabindex="-1"></a>         <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>))) <span class="sc">+</span></span>
<span id="cb95-53"><a href="#cb95-53" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">theme_pubr</span>(<span class="at">base_size =</span> <span class="dv">20</span>, <span class="at">legend =</span> <span class="st">"bottom"</span>, <span class="at">x.text.angle =</span> <span class="dv">90</span>)</span>
<span id="cb95-54"><a href="#cb95-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-55"><a href="#cb95-55" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">ggsave2</span>(<span class="fu">file.path</span>(dir_save, <span class="st">'p_bar_cell_STIM_sctype.png'</span>), p_bar_cell,</span>
<span id="cb95-56"><a href="#cb95-56" aria-hidden="true" tabindex="-1"></a>                 <span class="at">units =</span> <span class="st">'cm'</span>,</span>
<span id="cb95-57"><a href="#cb95-57" aria-hidden="true" tabindex="-1"></a>                 <span class="at">width =</span> <span class="dv">15</span>,</span>
<span id="cb95-58"><a href="#cb95-58" aria-hidden="true" tabindex="-1"></a>                 <span class="at">height =</span> <span class="dv">45</span>)</span>
<span id="cb95-59"><a href="#cb95-59" aria-hidden="true" tabindex="-1"></a>p_bar_cell</span>
<span id="cb95-60"><a href="#cb95-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-61"><a href="#cb95-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-62"><a href="#cb95-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-63"><a href="#cb95-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-64"><a href="#cb95-64" aria-hidden="true" tabindex="-1"></a><span class="co">#RNA_snn_res.1.2</span></span>
<span id="cb95-65"><a href="#cb95-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the total number of cells per sample</span></span>
<span id="cb95-66"><a href="#cb95-66" aria-hidden="true" tabindex="-1"></a>total_cells_per_sample <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb95-67"><a href="#cb95-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(VISIT, PID, RNA_snn_res.<span class="fl">1.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-68"><a href="#cb95-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_cells =</span> <span class="fu">n</span>())</span>
<span id="cb95-69"><a href="#cb95-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-70"><a href="#cb95-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the number of cells per cluster per sample</span></span>
<span id="cb95-71"><a href="#cb95-71" aria-hidden="true" tabindex="-1"></a>cells_per_cluster_sample <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb95-72"><a href="#cb95-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(STIM, VISIT, PID, RNA_snn_res.<span class="fl">0.6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-73"><a href="#cb95-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">cluster_cells =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb95-74"><a href="#cb95-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb95-75"><a href="#cb95-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-76"><a href="#cb95-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the proportion of cells in each cluster for each sample</span></span>
<span id="cb95-77"><a href="#cb95-77" aria-hidden="true" tabindex="-1"></a>proportions <span class="ot">&lt;-</span> cells_per_cluster_sample <span class="sc">%&gt;%</span></span>
<span id="cb95-78"><a href="#cb95-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(total_cells_per_sample, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"PID"</span>, <span class="st">"RNA_snn_res.0.6"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-79"><a href="#cb95-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">proportion =</span> cluster_cells <span class="sc">/</span> total_cells) <span class="sc">%&gt;%</span></span>
<span id="cb95-80"><a href="#cb95-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(STIM, VISIT, PID, RNA_snn_res.<span class="fl">0.6</span>, proportion)</span>
<span id="cb95-81"><a href="#cb95-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-82"><a href="#cb95-82" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot the data to create a matrix</span></span>
<span id="cb95-83"><a href="#cb95-83" aria-hidden="true" tabindex="-1"></a>proportion_df <span class="ot">&lt;-</span> proportions <span class="sc">%&gt;%</span></span>
<span id="cb95-84"><a href="#cb95-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> RNA_snn_res.<span class="fl">0.6</span>, <span class="at">values_from =</span> proportion)</span>
<span id="cb95-85"><a href="#cb95-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-86"><a href="#cb95-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to matrix</span></span>
<span id="cb95-87"><a href="#cb95-87" aria-hidden="true" tabindex="-1"></a><span class="co">#proportion_matrix &lt;- as.matrix(proportion_df [,-1])</span></span>
<span id="cb95-88"><a href="#cb95-88" aria-hidden="true" tabindex="-1"></a><span class="co">#rownames(proportion_matrix) &lt;- proportion_df$SampleName</span></span>
<span id="cb95-89"><a href="#cb95-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-90"><a href="#cb95-90" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> proportion_df <span class="sc">%&gt;%</span> </span>
<span id="cb95-91"><a href="#cb95-91" aria-hidden="true" tabindex="-1"></a>  <span class="co">#as.data.frame() %&gt;% </span></span>
<span id="cb95-92"><a href="#cb95-92" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tibble::rownames_to_column(var = "group") %&gt;% </span></span>
<span id="cb95-93"><a href="#cb95-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">names_to =</span> <span class="st">"RNA_snn_res.0.6"</span>, <span class="at">values_to =</span> <span class="st">"Prop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb95-94"><a href="#cb95-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA_snn_res.0.6 =</span> <span class="fu">as.character</span>(RNA_snn_res.<span class="fl">0.6</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-95"><a href="#cb95-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(cluster_names, <span class="at">by =</span> <span class="st">"RNA_snn_res.0.6"</span>) <span class="co">#%&gt;% </span></span>
<span id="cb95-96"><a href="#cb95-96" aria-hidden="true" tabindex="-1"></a>  <span class="co">#select(-RNA_snn_res.0.6) #%&gt;% </span></span>
<span id="cb95-97"><a href="#cb95-97" aria-hidden="true" tabindex="-1"></a>  <span class="co">#pivot_wider(names_from = "cell_type", values_from = "Prop")</span></span>
<span id="cb95-98"><a href="#cb95-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-99"><a href="#cb95-99" aria-hidden="true" tabindex="-1"></a>p_bar_cell <span class="ot">&lt;-</span> test <span class="sc">%&gt;%</span> </span>
<span id="cb95-100"><a href="#cb95-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> RNA_snn_res.<span class="fl">0.6</span>, <span class="at">y =</span> Prop, <span class="at">fill =</span> STIM)) <span class="sc">+</span></span>
<span id="cb95-101"><a href="#cb95-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>),</span>
<span id="cb95-102"><a href="#cb95-102" aria-hidden="true" tabindex="-1"></a>           <span class="at">stat =</span> <span class="st">"summary"</span>,</span>
<span id="cb95-103"><a href="#cb95-103" aria-hidden="true" tabindex="-1"></a>           <span class="at">fun =</span> <span class="st">"median"</span>,</span>
<span id="cb95-104"><a href="#cb95-104" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb95-105"><a href="#cb95-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(PID <span class="sc">~</span> VISIT) <span class="sc">+</span>  <span class="co"># This creates separate plots for each VISIT value</span></span>
<span id="cb95-106"><a href="#cb95-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Proportion of cells"</span>) <span class="sc">+</span></span>
<span id="cb95-107"><a href="#cb95-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb95-108"><a href="#cb95-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">alpha =</span> <span class="st">"none"</span>,</span>
<span id="cb95-109"><a href="#cb95-109" aria-hidden="true" tabindex="-1"></a>         <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>))) <span class="sc">+</span></span>
<span id="cb95-110"><a href="#cb95-110" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">theme_pubr</span>(<span class="at">base_size =</span> <span class="dv">20</span>, <span class="at">legend =</span> <span class="st">"bottom"</span>, <span class="at">x.text.angle =</span> <span class="dv">90</span>)</span>
<span id="cb95-111"><a href="#cb95-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-112"><a href="#cb95-112" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot</span></span>
<span id="cb95-113"><a href="#cb95-113" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">ggsave2</span>(<span class="fu">file.path</span>(dir_save, <span class="st">'p_bar_cell_STIM_RNA_snn_res.0.6.png'</span>), p_bar_cell,</span>
<span id="cb95-114"><a href="#cb95-114" aria-hidden="true" tabindex="-1"></a>                 <span class="at">units =</span> <span class="st">'cm'</span>,</span>
<span id="cb95-115"><a href="#cb95-115" aria-hidden="true" tabindex="-1"></a>                 <span class="at">width =</span> <span class="dv">30</span>,</span>
<span id="cb95-116"><a href="#cb95-116" aria-hidden="true" tabindex="-1"></a>                 <span class="at">height =</span> <span class="dv">20</span>)</span>
<span id="cb95-117"><a href="#cb95-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-118"><a href="#cb95-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-119"><a href="#cb95-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-120"><a href="#cb95-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-121"><a href="#cb95-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-122"><a href="#cb95-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-123"><a href="#cb95-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-124"><a href="#cb95-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-125"><a href="#cb95-125" aria-hidden="true" tabindex="-1"></a><span class="co">#sctype</span></span>
<span id="cb95-126"><a href="#cb95-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-127"><a href="#cb95-127" aria-hidden="true" tabindex="-1"></a>cluster_names <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb95-128"><a href="#cb95-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(VISIT_STIM)</span>
<span id="cb95-129"><a href="#cb95-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-130"><a href="#cb95-130" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(cluster_names) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb95-131"><a href="#cb95-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-132"><a href="#cb95-132" aria-hidden="true" tabindex="-1"></a>cluster_names <span class="ot">&lt;-</span> cluster_names <span class="sc">%&gt;%</span> </span>
<span id="cb95-133"><a href="#cb95-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb95-134"><a href="#cb95-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-135"><a href="#cb95-135" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the total number of cells per sample</span></span>
<span id="cb95-136"><a href="#cb95-136" aria-hidden="true" tabindex="-1"></a>total_cells_per_sample <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb95-137"><a href="#cb95-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(VISIT_STIM) <span class="sc">%&gt;%</span></span>
<span id="cb95-138"><a href="#cb95-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_cells =</span> <span class="fu">n</span>())</span>
<span id="cb95-139"><a href="#cb95-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-140"><a href="#cb95-140" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the number of cells per cluster per sample</span></span>
<span id="cb95-141"><a href="#cb95-141" aria-hidden="true" tabindex="-1"></a>cells_per_cluster_sample <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb95-142"><a href="#cb95-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(VISIT_STIM, sctype_classification) <span class="sc">%&gt;%</span></span>
<span id="cb95-143"><a href="#cb95-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">cluster_cells =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb95-144"><a href="#cb95-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb95-145"><a href="#cb95-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-146"><a href="#cb95-146" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the proportion of cells in each cluster for each sample</span></span>
<span id="cb95-147"><a href="#cb95-147" aria-hidden="true" tabindex="-1"></a>proportions <span class="ot">&lt;-</span> cells_per_cluster_sample <span class="sc">%&gt;%</span></span>
<span id="cb95-148"><a href="#cb95-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(total_cells_per_sample, <span class="at">by =</span> <span class="st">"VISIT_STIM"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-149"><a href="#cb95-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">proportion =</span> cluster_cells <span class="sc">/</span> total_cells) <span class="sc">%&gt;%</span></span>
<span id="cb95-150"><a href="#cb95-150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(VISIT_STIM, sctype_classification, proportion)</span>
<span id="cb95-151"><a href="#cb95-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-152"><a href="#cb95-152" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot the data to create a matrix</span></span>
<span id="cb95-153"><a href="#cb95-153" aria-hidden="true" tabindex="-1"></a>proportion_df <span class="ot">&lt;-</span> proportions <span class="sc">%&gt;%</span></span>
<span id="cb95-154"><a href="#cb95-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> VISIT_STIM, <span class="at">values_from =</span> proportion)</span>
<span id="cb95-155"><a href="#cb95-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-156"><a href="#cb95-156" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to matrix</span></span>
<span id="cb95-157"><a href="#cb95-157" aria-hidden="true" tabindex="-1"></a><span class="co">#proportion_matrix &lt;- as.matrix(proportion_df [,-1])</span></span>
<span id="cb95-158"><a href="#cb95-158" aria-hidden="true" tabindex="-1"></a><span class="co">#rownames(proportion_matrix) &lt;- proportion_df$SampleName</span></span>
<span id="cb95-159"><a href="#cb95-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-160"><a href="#cb95-160" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> proportion_df <span class="sc">%&gt;%</span> </span>
<span id="cb95-161"><a href="#cb95-161" aria-hidden="true" tabindex="-1"></a>  <span class="co">#as.data.frame() %&gt;% </span></span>
<span id="cb95-162"><a href="#cb95-162" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tibble::rownames_to_column(var = "group") %&gt;% </span></span>
<span id="cb95-163"><a href="#cb95-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">names_to =</span> <span class="st">"VISIT_STIM"</span>, <span class="at">values_to =</span> <span class="st">"Prop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb95-164"><a href="#cb95-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">VISIT_STIM =</span> <span class="fu">as.character</span>(VISIT_STIM)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-165"><a href="#cb95-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(cluster_names, <span class="at">by =</span> <span class="st">"VISIT_STIM"</span>) <span class="co">#%&gt;% </span></span>
<span id="cb95-166"><a href="#cb95-166" aria-hidden="true" tabindex="-1"></a>  <span class="co">#select(-RNA_snn_res.0.6) #%&gt;% </span></span>
<span id="cb95-167"><a href="#cb95-167" aria-hidden="true" tabindex="-1"></a>  <span class="co">#pivot_wider(names_from = "cell_type", values_from = "Prop")</span></span>
<span id="cb95-168"><a href="#cb95-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-169"><a href="#cb95-169" aria-hidden="true" tabindex="-1"></a>p_bar_cell <span class="ot">&lt;-</span> test <span class="sc">%&gt;%</span> </span>
<span id="cb95-170"><a href="#cb95-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> VISIT_STIM, <span class="at">y =</span> Prop, <span class="at">fill =</span> sctype_classification)) <span class="sc">+</span></span>
<span id="cb95-171"><a href="#cb95-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>),</span>
<span id="cb95-172"><a href="#cb95-172" aria-hidden="true" tabindex="-1"></a>           <span class="at">stat =</span> <span class="st">"summary"</span>,</span>
<span id="cb95-173"><a href="#cb95-173" aria-hidden="true" tabindex="-1"></a>           <span class="at">fun =</span> <span class="st">"median"</span>,</span>
<span id="cb95-174"><a href="#cb95-174" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb95-175"><a href="#cb95-175" aria-hidden="true" tabindex="-1"></a>  <span class="co">#facet_wrap(~ VISIT) +  # This creates separate plots for each VISIT value</span></span>
<span id="cb95-176"><a href="#cb95-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Proportion of cells"</span>) <span class="sc">+</span></span>
<span id="cb95-177"><a href="#cb95-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb95-178"><a href="#cb95-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">alpha =</span> <span class="st">"none"</span>,</span>
<span id="cb95-179"><a href="#cb95-179" aria-hidden="true" tabindex="-1"></a>         <span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>))) <span class="sc">+</span></span>
<span id="cb95-180"><a href="#cb95-180" aria-hidden="true" tabindex="-1"></a>  ggpubr<span class="sc">::</span><span class="fu">theme_pubr</span>(<span class="at">base_size =</span> <span class="dv">20</span>, <span class="at">legend =</span> <span class="st">"bottom"</span>, <span class="at">x.text.angle =</span> <span class="dv">90</span>)</span>
<span id="cb95-181"><a href="#cb95-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-182"><a href="#cb95-182" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot</span></span>
<span id="cb95-183"><a href="#cb95-183" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">ggsave2</span>(<span class="fu">file.path</span>(dir_save, <span class="st">'p_bar_cell_STIM_sctype.png'</span>), p_bar_cell,</span>
<span id="cb95-184"><a href="#cb95-184" aria-hidden="true" tabindex="-1"></a>                 <span class="at">units =</span> <span class="st">'cm'</span>,</span>
<span id="cb95-185"><a href="#cb95-185" aria-hidden="true" tabindex="-1"></a>                 <span class="at">width =</span> <span class="dv">30</span>,</span>
<span id="cb95-186"><a href="#cb95-186" aria-hidden="true" tabindex="-1"></a>                 <span class="at">height =</span> <span class="dv">20</span>)</span>
<span id="cb95-187"><a href="#cb95-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-188"><a href="#cb95-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-189"><a href="#cb95-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-190"><a href="#cb95-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-191"><a href="#cb95-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-192"><a href="#cb95-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-193"><a href="#cb95-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-194"><a href="#cb95-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-195"><a href="#cb95-195" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################</span></span>
<span id="cb95-196"><a href="#cb95-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract metadata</span></span>
<span id="cb95-197"><a href="#cb95-197" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> obj<span class="sc">@</span>meta.data</span>
<span id="cb95-198"><a href="#cb95-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-199"><a href="#cb95-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique clusters and stimulation groups</span></span>
<span id="cb95-200"><a href="#cb95-200" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">unique</span>(metadata<span class="sc">$</span>seurat_clusters)</span>
<span id="cb95-201"><a href="#cb95-201" aria-hidden="true" tabindex="-1"></a>stim_groups <span class="ot">&lt;-</span> <span class="fu">unique</span>(metadata<span class="sc">$</span>STIM)</span>
<span id="cb95-202"><a href="#cb95-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-203"><a href="#cb95-203" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store results</span></span>
<span id="cb95-204"><a href="#cb95-204" aria-hidden="true" tabindex="-1"></a>results_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb95-205"><a href="#cb95-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-206"><a href="#cb95-206" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over each stimulation group and cluster to calculate the metrics</span></span>
<span id="cb95-207"><a href="#cb95-207" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (stim <span class="cf">in</span> stim_groups) {</span>
<span id="cb95-208"><a href="#cb95-208" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (cluster <span class="cf">in</span> clusters) {</span>
<span id="cb95-209"><a href="#cb95-209" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Subset metadata for the current stim and cluster</span></span>
<span id="cb95-210"><a href="#cb95-210" aria-hidden="true" tabindex="-1"></a>    subset_metadata <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb95-211"><a href="#cb95-211" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(seurat_clusters <span class="sc">==</span> cluster, STIM <span class="sc">==</span> stim)</span>
<span id="cb95-212"><a href="#cb95-212" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb95-213"><a href="#cb95-213" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate metrics</span></span>
<span id="cb95-214"><a href="#cb95-214" aria-hidden="true" tabindex="-1"></a>    mapped_reads <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_metadata<span class="sc">$</span>reads.mapped, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-215"><a href="#cb95-215" aria-hidden="true" tabindex="-1"></a>    exon_reads <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_metadata<span class="sc">$</span>ExonReads, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-216"><a href="#cb95-216" aria-hidden="true" tabindex="-1"></a>    intron_reads <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_metadata<span class="sc">$</span>IntronReads, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-217"><a href="#cb95-217" aria-hidden="true" tabindex="-1"></a>    nTrans <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_metadata<span class="sc">$</span>nCount_RNA, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-218"><a href="#cb95-218" aria-hidden="true" tabindex="-1"></a>    nGenes <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_metadata<span class="sc">$</span>nFeature_RNA, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-219"><a href="#cb95-219" aria-hidden="true" tabindex="-1"></a>    percMito <span class="ot">&lt;-</span> <span class="fu">mean</span>(subset_metadata<span class="sc">$</span>percent.mito, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb95-220"><a href="#cb95-220" aria-hidden="true" tabindex="-1"></a>    nCells <span class="ot">&lt;-</span> <span class="fu">nrow</span>(subset_metadata)</span>
<span id="cb95-221"><a href="#cb95-221" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb95-222"><a href="#cb95-222" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create a data frame with the calculated metrics</span></span>
<span id="cb95-223"><a href="#cb95-223" aria-hidden="true" tabindex="-1"></a>    metrics_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb95-224"><a href="#cb95-224" aria-hidden="true" tabindex="-1"></a>      <span class="at">Stimulation =</span> stim,</span>
<span id="cb95-225"><a href="#cb95-225" aria-hidden="true" tabindex="-1"></a>      <span class="at">Cluster =</span> cluster,</span>
<span id="cb95-226"><a href="#cb95-226" aria-hidden="true" tabindex="-1"></a>      <span class="at">MappedReads =</span> mapped_reads,</span>
<span id="cb95-227"><a href="#cb95-227" aria-hidden="true" tabindex="-1"></a>      <span class="at">ExonReads =</span> exon_reads,</span>
<span id="cb95-228"><a href="#cb95-228" aria-hidden="true" tabindex="-1"></a>      <span class="at">IntronReads =</span> intron_reads,</span>
<span id="cb95-229"><a href="#cb95-229" aria-hidden="true" tabindex="-1"></a>      <span class="at">nTrans =</span> nTrans,</span>
<span id="cb95-230"><a href="#cb95-230" aria-hidden="true" tabindex="-1"></a>      <span class="at">nGenes =</span> nGenes,</span>
<span id="cb95-231"><a href="#cb95-231" aria-hidden="true" tabindex="-1"></a>      <span class="at">percMito =</span> percMito,</span>
<span id="cb95-232"><a href="#cb95-232" aria-hidden="true" tabindex="-1"></a>      <span class="at">nCells =</span> nCells</span>
<span id="cb95-233"><a href="#cb95-233" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb95-234"><a href="#cb95-234" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb95-235"><a href="#cb95-235" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Append the data frame to the results list</span></span>
<span id="cb95-236"><a href="#cb95-236" aria-hidden="true" tabindex="-1"></a>    results_list[[<span class="fu">paste0</span>(<span class="st">"Cluster"</span>, cluster, <span class="st">"_Stim"</span>, stim)]] <span class="ot">&lt;-</span> metrics_df</span>
<span id="cb95-237"><a href="#cb95-237" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb95-238"><a href="#cb95-238" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-239"><a href="#cb95-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-240"><a href="#cb95-240" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all results into a single data frame</span></span>
<span id="cb95-241"><a href="#cb95-241" aria-hidden="true" tabindex="-1"></a>summary_table <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(results_list, <span class="at">.id =</span> <span class="st">"Cluster_Stim"</span>)</span>
<span id="cb95-242"><a href="#cb95-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-243"><a href="#cb95-243" aria-hidden="true" tabindex="-1"></a><span class="co"># Reshape the summary table to have clusters and stimulations as columns</span></span>
<span id="cb95-244"><a href="#cb95-244" aria-hidden="true" tabindex="-1"></a>summary_table_wide <span class="ot">&lt;-</span> summary_table <span class="sc">%&gt;%</span> </span>
<span id="cb95-245"><a href="#cb95-245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Cluster_Stim, <span class="at">values_from =</span> <span class="fu">c</span>(MappedReads, ExonReads, IntronReads, nTrans, nGenes, percMito, nCells))</span>
<span id="cb95-246"><a href="#cb95-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-247"><a href="#cb95-247" aria-hidden="true" tabindex="-1"></a><span class="co"># View the summary table</span></span>
<span id="cb95-248"><a href="#cb95-248" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(summary_table_wide)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="identify-conserved-cell-type-markers" class="level3">
<h3 class="anchored" data-anchor-id="identify-conserved-cell-type-markers">Identify conserved cell type markers</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Idents(obj) &lt;- "sctype_classification"</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list of all sctype classifications in obj</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sctype.types &lt;- levels(Idents(obj))</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store cluster markers</span></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a><span class="co">#cluster.markers &lt;- list()</span></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each Seurat cluster in obj</span></span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a><span class="co">#for (sctype.type in sctype.types) {</span></span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#sctype.markers &lt;- FindConservedMarkers(obj, ident.1 = sctype.type, grouping.var = "group", verbose = FALSE)</span></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify differentially expressed genes</span></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="overlay-expression-of-adts-on-rna-clusters" class="level3">
<h3 class="anchored" data-anchor-id="overlay-expression-of-adts-on-rna-clusters">Overlay expression of ADTs on RNA clusters</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list of all cluster identities in obj</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>cluster.ids <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">Idents</span>(obj))</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store cluster markers</span></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>cluster.markers.RNAandADT <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize the ADT data</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(obj) <span class="ot">&lt;-</span> <span class="st">"ADT"</span></span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(obj, <span class="at">normalization.method =</span> <span class="st">"CLR"</span>, <span class="at">margin =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing across cells</code></pre>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each Seurat cluster in obj</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cluster.id <span class="cf">in</span> cluster.ids) {</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find genes for RNA assay sorted by adjusted p value </span></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>  genes_RNA <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(obj, <span class="at">ident.1 =</span> cluster.id, <span class="at">assay =</span> <span class="st">"RNA"</span>)</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>  genes_RNA <span class="ot">&lt;-</span> <span class="fu">rownames</span>(genes_RNA[<span class="fu">order</span>(genes_RNA<span class="sc">$</span>p_val_adj), ])</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find genes for ADT assay sorted by adjusted p value </span></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>  genes_ADT <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(obj, <span class="at">ident.1 =</span> cluster.id, <span class="at">assay =</span> <span class="st">"ADT"</span>)</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>  genes_ADT <span class="ot">&lt;-</span> <span class="fu">rownames</span>(genes_ADT[<span class="fu">order</span>(genes_ADT<span class="sc">$</span>p_val_adj), ])</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find common genes between RNA and ADT assays</span></span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>  cluster.markers.RNAandADT <span class="ot">&lt;-</span> <span class="fu">intersect</span>(genes_RNA, genes_ADT)</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>  object.name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster.markers."</span>, cluster.id)</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot violin plots of relative expression </span></span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DefaultAssay</span>(obj) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a>  vln_RNA <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">features =</span> <span class="fu">c</span>(cluster.markers.RNAandADT)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">log =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pt.size =</span> <span class="dv">0</span>) </span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a>  main_title <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> </span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="fu">paste0</span>(<span class="st">"RNA Cluster "</span>, cluster.id), <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a>  vln_RNA <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(main_title, vln_RNA, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>))</span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DefaultAssay</span>(obj) <span class="ot">&lt;-</span> <span class="st">"ADT"</span></span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a>  vln_ADT <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a>                      <span class="at">features =</span> <span class="fu">c</span>(cluster.markers.RNAandADT)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a>                      <span class="at">log =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb99-35"><a href="#cb99-35" aria-hidden="true" tabindex="-1"></a>  main_title <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> </span>
<span id="cb99-36"><a href="#cb99-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="fu">paste0</span>(<span class="st">"ADT Cluster "</span>, cluster.id), <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb99-37"><a href="#cb99-37" aria-hidden="true" tabindex="-1"></a>  vln_ADT <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(main_title, vln_ADT, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>))</span>
<span id="cb99-38"><a href="#cb99-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-39"><a href="#cb99-39" aria-hidden="true" tabindex="-1"></a>  vln_plots <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Multimodal_DEG_VlnPlotRelExp_Top10_"</span>, object.name, <span class="st">".pdf"</span>)</span>
<span id="cb99-40"><a href="#cb99-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, vln_plots), <span class="at">height =</span> <span class="dv">30</span>, <span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb99-41"><a href="#cb99-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(vln_RNA <span class="sc">|</span> vln_ADT)</span>
<span id="cb99-42"><a href="#cb99-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb99-43"><a href="#cb99-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-44"><a href="#cb99-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot feature plots overlaid onto UMAP</span></span>
<span id="cb99-45"><a href="#cb99-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-46"><a href="#cb99-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DefaultAssay</span>(obj) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb99-47"><a href="#cb99-47" aria-hidden="true" tabindex="-1"></a>  feature_RNA <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(obj,</span>
<span id="cb99-48"><a href="#cb99-48" aria-hidden="true" tabindex="-1"></a>                              <span class="at">features =</span> <span class="fu">c</span>(cluster.markers.RNAandADT)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb99-49"><a href="#cb99-49" aria-hidden="true" tabindex="-1"></a>                              <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-50"><a href="#cb99-50" aria-hidden="true" tabindex="-1"></a>                              <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb99-51"><a href="#cb99-51" aria-hidden="true" tabindex="-1"></a>                              <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb99-52"><a href="#cb99-52" aria-hidden="true" tabindex="-1"></a>                              <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb99-53"><a href="#cb99-53" aria-hidden="true" tabindex="-1"></a>  main_title <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> </span>
<span id="cb99-54"><a href="#cb99-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="fu">paste0</span>(<span class="st">"RNA Cluster "</span>, cluster.id), <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb99-55"><a href="#cb99-55" aria-hidden="true" tabindex="-1"></a>  feature_RNA <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(main_title, feature_RNA, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>))</span>
<span id="cb99-56"><a href="#cb99-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-57"><a href="#cb99-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DefaultAssay</span>(obj) <span class="ot">&lt;-</span> <span class="st">"ADT"</span></span>
<span id="cb99-58"><a href="#cb99-58" aria-hidden="true" tabindex="-1"></a>  feature_ADT <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(obj,</span>
<span id="cb99-59"><a href="#cb99-59" aria-hidden="true" tabindex="-1"></a>                              <span class="at">features =</span> <span class="fu">c</span>(cluster.markers.RNAandADT)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb99-60"><a href="#cb99-60" aria-hidden="true" tabindex="-1"></a>                              <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-61"><a href="#cb99-61" aria-hidden="true" tabindex="-1"></a>                              <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb99-62"><a href="#cb99-62" aria-hidden="true" tabindex="-1"></a>                              <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb99-63"><a href="#cb99-63" aria-hidden="true" tabindex="-1"></a>                              <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb99-64"><a href="#cb99-64" aria-hidden="true" tabindex="-1"></a>  main_title <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> </span>
<span id="cb99-65"><a href="#cb99-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="fu">paste0</span>(<span class="st">"ADT Cluster "</span>, cluster.id), <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb99-66"><a href="#cb99-66" aria-hidden="true" tabindex="-1"></a>  feature_ADT <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(main_title, feature_ADT, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>))</span>
<span id="cb99-67"><a href="#cb99-67" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-68"><a href="#cb99-68" aria-hidden="true" tabindex="-1"></a>  feature_plots <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Multimodal_DEG_FeaturePlot_Top10_"</span>, object.name, <span class="st">".pdf"</span>)</span>
<span id="cb99-69"><a href="#cb99-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, feature_plots), <span class="at">height =</span> <span class="dv">30</span>, <span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb99-70"><a href="#cb99-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(feature_RNA <span class="sc">|</span> feature_ADT)</span>
<span id="cb99-71"><a href="#cb99-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb99-72"><a href="#cb99-72" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(obj, <span class="at">file =</span> <span class="fu">paste0</span>(dir_save, <span class="st">"Seurat_ClusterIDCellsRNAonly_obj.Rdata"</span>))</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(obj, <span class="at">file =</span> <span class="fu">paste0</span>(dir_save, <span class="st">"Seurat_ClusterIDCellsRNAonly_obj.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="adt-protein" class="level2">
<h2 class="anchored" data-anchor-id="adt-protein">ADT (Protein)</h2>
<section id="set-default-assay-to-adt" class="level3">
<h3 class="anchored" data-anchor-id="set-default-assay-to-adt">Set default assay to ADT</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(obj) <span class="ot">&lt;-</span> <span class="st">"ADT"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="split-the-object-into-multiple-layers-for-integration-based-on-adt-assay" class="level3">
<h3 class="anchored" data-anchor-id="split-the-object-into-multiple-layers-for-integration-based-on-adt-assay">Split the object into multiple layers for integration based on ADT Assay</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>obj[[<span class="st">"ADT"</span>]] <span class="ot">&lt;-</span> <span class="fu">split</span>(obj[[<span class="st">"ADT"</span>]], <span class="at">f =</span> obj<span class="sc">$</span>RUN)</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>obj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
54869 features across 66972 samples within 3 assays 
Active assay: ADT (165 features, 0 variable features)
 6 layers present: counts.M72RUN001, counts.M72RUN002, counts.M72RUN005, data.M72RUN001, data.M72RUN002, data.M72RUN005
 2 other assays present: RNA, HTO
 4 dimensional reductions calculated: pca, umap.unintegrated, integrated.cca, umap</code></pre>
</div>
</div>
</section>
<section id="normalize-the-adt-data" class="level3">
<h3 class="anchored" data-anchor-id="normalize-the-adt-data">Normalize the ADT data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(obj, <span class="at">normalization.method =</span> <span class="st">"CLR"</span>, <span class="at">margin =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="identify-variable-features-1" class="level3">
<h3 class="anchored" data-anchor-id="identify-variable-features-1">Identify variable features</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(obj, <span class="at">selection.method =</span> <span class="st">"vst"</span>)</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>top10.adt <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(obj), <span class="dv">10</span>)</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a><span class="co"># plot variable features and label top 10 variable features</span></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(obj)</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span><span class="fu">LabelPoints</span>(<span class="at">plot =</span> plot3, <span class="at">points =</span> top10.adt, <span class="at">repel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>plot4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="scale-the-adt-data" class="level3">
<h3 class="anchored" data-anchor-id="scale-the-adt-data">Scale the ADT data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering and scaling data matrix</code></pre>
</div>
</div>
</section>
<section id="perform-linear-dimensional-reduction-1" class="level3">
<h3 class="anchored" data-anchor-id="perform-linear-dimensional-reduction-1">Perform linear dimensional reduction</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA for linear dimensional reduction on scaled data. Give reduction a new name to avoid overwriting RNA reduction. </span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(obj, <span class="at">features =</span> <span class="cn">NULL</span>, <span class="at">reduction.name =</span> <span class="st">'pca.adt'</span>) <span class="co"># VariableFeatures(object = obj)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  CD74, CD23, CD83, IgE, CD303, LOX.1, CD223, CD24, CD79b, CD134 
       CD200, CD142, FceRIa, CD163, CX3CR1, CD112, CD131, Isotype-MOPC.173, CD154, Podoplanin 
       CD169, CD279, CD107a, CD115, CD141, CD152, Isotype-G0114F7, CD122, CD109, Isotype-RTK4174 
Negative:  CD48, CD162, CD5, CD2, CD45-HI30, CD11a, HuMs.CD44, CD47, CD71, CD49d 
       CD62L, CD99, CD28, CD3-UCHT1, CD27, CD55, CD29, CD127, CD43, HuMsRt.CD278 
       CD7, CD18, HuMs.CD49f, CD352, CD4-RPA.T4, HuMs.integrin.b7, CD54, CD45RA, CD26, CD31 
PC_ 2 
Positive:  CD3-UCHT1, TCR.AB, CD28, CD7, CD2, CD52, CD4-RPA.T4, CD183, CD127, CD352 
       CD5, HuMsRt.CD278, CD27, CD279, CD99, TIGIT, CD45RO, CD162, CD62L, CD26 
       CD314, CD122, CD43, CD134, CD223, CD270, KLRG1, CD303, CD23, CD8 
Negative:  CD33, CD11c, HLA.DR, CD54, HLA.DR.DP.DQ, CD304, CD13, CLEC12A, CD11b, CD63 
       CD319, CD123, CD39, CD32, CD86, CD172a, CD35, CD116, CD88, CD328 
       CD93, CD64, CD155, CD36, CD274, CD31, CD62P, CD9, CD41, CD61 
PC_ 3 
Positive:  CD244, CD16, CD43, CD56, CD328, CD62P, GPR56, CD94, CD18, CD226-11A8 
       CD11c, CD151, CD11b, CD158, CD29, CD57, CD335, CD172a, CD13, NKp80 
       CLEC1B, CD162, TIGIT, CD61, KLRG1, CD95, CD11a, CLEC12A, CD41, CD33 
Negative:  CD72, CD19, CD268, CD21, CD22, IgD, CD20-2H7, CD35, CD73, CD272 
       CD275-B7.RP1, IgM, CD37, CD40, CD55, CD185, HLA.DR.DP.DQ, CD196, HLA.DR, CD1c 
       CD32, CD82, CD200, CD150, CD62L, Ig.LightChain.k, CD39, Ig.LightChain.l, HuMs.integrin.b7, CD27 
PC_ 4 
Positive:  CD16, CD45RA, CD244, CD56, GPR56, CD94, CD57, CD158, CD158e1, NKp80 
       CD335, CD38-HIT2, CD305-LAIR1, CD72, CD43, CD22, CD19, CD268, CD21, IgD 
       CD20-2H7, CD352, CD8, CD49d, CD328, KLRG1, CD81, TIGIT, CD11a, Ig.LightChain.k 
Negative:  CD45RO, CD4-RPA.T4, CD28, CD150, CD127, CD194, TCR.AB, CD26, CD3-UCHT1, HuMs.CD49f 
       CD5, CD95, HuMsRt.CD278, CD82, CD274, CD58, CD25, CD27, CD93, CD49a 
       CD55, CD88, CD141, CD196, CD84, CD13, CD107a, CD192, CD279, CD134 
PC_ 5 
Positive:  CD81, CD95, CD11a, CD150, CD18, CD58, CD48, CD45-HI30, HuMs.CD44, CD274 
       CD328, CD161, HLA.ABC, CD319, CD88, CD270, CD352, CD49d, CD45RO, CD305-LAIR1 
       CD119, CD162, CD49a, CD11b, CD85j, CD11c, CD13, CD39, KLRG1, NKp80 
Negative:  CD61, CLEC1B, CD9, CD49b, CD41, CD42b, CD62P, CD36, HuMs.CD49f, CD151 
       CD31, CD226-11A8, CD84, CD71, CD73, CD69, CD22, CD19, CD72, CD194 
       CD45RA, CD21, CD20-2H7, CD63, CD79b, CD40, IgD, Isotype-RTK2758, CD74, CD66b </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Key 'PC_' taken, using 'pcaadt_' instead</code></pre>
</div>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine and visualize PCA results a few different ways</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(obj[[<span class="st">"pca.adt"</span>]], <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">nfeatures =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pcaadt_ 1 
Positive:  CD74, CD23, CD83, IgE, CD303 
Negative:  CD48, CD162, CD5, CD2, CD45-HI30 
pcaadt_ 2 
Positive:  CD3-UCHT1, TCR.AB, CD28, CD7, CD2 
Negative:  CD33, CD11c, HLA.DR, CD54, HLA.DR.DP.DQ 
pcaadt_ 3 
Positive:  CD244, CD16, CD43, CD56, CD328 
Negative:  CD72, CD19, CD268, CD21, CD22 
pcaadt_ 4 
Positive:  CD16, CD45RA, CD244, CD56, GPR56 
Negative:  CD45RO, CD4-RPA.T4, CD28, CD150, CD127 
pcaadt_ 5 
Positive:  CD81, CD95, CD11a, CD150, CD18 
Negative:  CD61, CLEC1B, CD9, CD49b, CD41 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VizDimLoadings</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">reduction =</span> <span class="st">"pca.adt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"pca.adt"</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-54-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimHeatmap</span>(obj, <span class="at">reduction =</span> <span class="st">"pca.adt"</span>, <span class="at">dims =</span> <span class="dv">1</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-54-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimHeatmap</span>(obj, <span class="at">reduction =</span> <span class="st">"pca.adt"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Determine the dimensionality of the dataset using an Elbow plot (more efficient alternative to JackStraw procedure).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"pca.adt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cluster-cells-based-on-adt" class="level3">
<h3 class="anchored" data-anchor-id="cluster-cells-based-on-adt">Cluster cells based on ADT</h3>
<p>Start with standard Seurat clustering method WITHOUT INTEGRATION (using ElbowPlot output to determine dims).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(obj, <span class="at">reduction =</span> <span class="st">"pca.adt"</span>) <span class="co"># dims = 1:12</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(obj, <span class="at">cluster.name =</span> <span class="st">"unintegrated_clusters_adt"</span>) <span class="co"># res = 1.2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2079478

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9084
Number of communities: 23
Elapsed time: 20 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">reduction =</span> <span class="st">"pca.adt"</span>, <span class="at">reduction.name =</span> <span class="st">"umap.unintegrated.adt"</span>)</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"seurat_clusters"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"PID"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-58-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-58-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.unintegrated.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT_STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-58-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="perform-integration-of-adt-assay" class="level3">
<h3 class="anchored" data-anchor-id="perform-integration-of-adt-assay">Perform integration of ADT Assay</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">IntegrateLayers</span>(<span class="at">object =</span> obj, </span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> CCAIntegration, </span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">orig.reduction =</span> <span class="st">"pca.adt"</span>, </span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">new.reduction =</span> <span class="st">"integrated.cca.adt"</span>, </span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Adding a dimensional reduction (integrated.cca.adt) without the
associated assay being present</code></pre>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>obj[[<span class="st">"ADT"</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(obj[[<span class="st">"ADT"</span>]])</span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(obj, <span class="at">reduction =</span> <span class="st">"integrated.cca.adt"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing SNN</code></pre>
</div>
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(obj, <span class="at">resolution =</span> <span class="fl">1.2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8595
Number of communities: 26
Elapsed time: 17 seconds</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1 singletons identified. 25 final clusters.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">reduction =</span> <span class="st">"integrated.cca.adt"</span>, <span class="at">reduction.name =</span> <span class="st">"umap.adt"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>00:11:37 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>00:11:37 Read 66972 rows and found 30 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>00:11:37 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>00:11:37 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
00:11:45 Writing NN index file to temp file /var/folders/_r/hqxyyq757rv6rj2qtz_x9zf80000gn/T//RtmphIEAjd/filebc233752e194
00:11:45 Searching Annoy index using 1 thread, search_k = 3000
00:12:14 Annoy recall = 100%
00:12:15 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
00:12:18 Initializing from normalized Laplacian + noise (using RSpectra)
00:12:21 Commencing optimization for 200 epochs, with 3214160 positive edges
00:13:01 Optimization finished</code></pre>
</div>
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"seurat_clusters"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-60-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"PID"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-60-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-60-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT_STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-60-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="determine-more-accurate-cluster-resolution-using-clustree-1" class="level3">
<h3 class="anchored" data-anchor-id="determine-more-accurate-cluster-resolution-using-clustree-1">Determine more accurate cluster resolution using clustree</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>resolution.range <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span><span class="fl">0.2</span>, <span class="at">to =</span> <span class="dv">2</span>, <span class="at">by =</span> <span class="fl">0.2</span>)</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> Seurat<span class="sc">::</span><span class="fu">FindClusters</span>(<span class="at">object =</span> obj, <span class="at">resolution =</span> resolution.range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9426
Number of communities: 13
Elapsed time: 19 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9212
Number of communities: 14
Elapsed time: 20 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9017
Number of communities: 18
Elapsed time: 19 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8855
Number of communities: 21
Elapsed time: 19 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8714
Number of communities: 25
Elapsed time: 19 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8595
Number of communities: 26
Elapsed time: 18 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8476
Number of communities: 26
Elapsed time: 18 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8372
Number of communities: 30
Elapsed time: 19 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8276
Number of communities: 31
Elapsed time: 16 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 66972
Number of edges: 2402326

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8194
Number of communities: 39
Elapsed time: 17 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>clustree <span class="ot">&lt;-</span> <span class="fu">clustree</span>(obj, <span class="at">prefix =</span> <span class="st">"ADT_snn_res."</span>)</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"clustree.all_adt.pdf"</span>))</span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a>clustree</span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>umap.<span class="dv">0</span>.<span class="fl">4.</span>adt <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"ADT_snn_res.0.4"</span>)</span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a>umap.<span class="dv">0</span>.<span class="fl">6.</span>adt <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"ADT_snn_res.0.6"</span>)</span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a>umap.<span class="dv">0</span>.<span class="fl">8.</span>adt <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"ADT_snn_res.0.8"</span>)</span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a>umap.<span class="fl">1.</span>adt <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"ADT_snn_res.1"</span>)</span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a>umap.<span class="dv">1</span>.<span class="fl">2.</span>adt <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">label =</span> T, <span class="at">group.by =</span> <span class="st">"ADT_snn_res.1.2"</span>)</span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-7"><a href="#cb150-7" aria-hidden="true" tabindex="-1"></a>snn_all.adt <span class="ot">&lt;-</span> (umap.<span class="dv">0</span>.<span class="fl">4.</span>adt <span class="sc">|</span> umap.<span class="dv">0</span>.<span class="fl">6.</span>adt) <span class="sc">/</span> (umap.<span class="dv">0</span>.<span class="fl">8.</span>adt <span class="sc">|</span> umap.<span class="fl">1.</span>adt) <span class="sc">/</span> umap.<span class="dv">1</span>.<span class="fl">2.</span>adt</span>
<span id="cb150-8"><a href="#cb150-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-9"><a href="#cb150-9" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_snn_compare_adt.pdf"</span>), <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">20</span>)</span>
<span id="cb150-10"><a href="#cb150-10" aria-hidden="true" tabindex="-1"></a>snn_all.adt</span>
<span id="cb150-11"><a href="#cb150-11" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a>snn_all.adt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(<span class="at">object =</span> obj) <span class="ot">&lt;-</span> <span class="st">"ADT_snn_res.0.6"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualize-clusters-1" class="level3">
<h3 class="anchored" data-anchor-id="visualize-clusters-1">Visualize clusters</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot UMAPs</span></span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_res.0.6_adt.pdf"</span>))</span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj,</span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.adt"</span>,</span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">6</span>)</span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allHIVES_res.0.6_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">25</span>)</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj,</span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.adt"</span>,</span>
<span id="cb156-4"><a href="#cb156-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb156-5"><a href="#cb156-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb156-6"><a href="#cb156-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb156-7"><a href="#cb156-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"HIVE"</span>,</span>
<span id="cb156-8"><a href="#cb156-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 12 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allPID_res.0.6_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">25</span>)</span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj,</span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.adt"</span>,</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"PID"</span>,</span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 6 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allSTIM_res.0.6_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj,</span>
<span id="cb164-3"><a href="#cb164-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.adt"</span>,</span>
<span id="cb164-4"><a href="#cb164-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb164-5"><a href="#cb164-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb164-6"><a href="#cb164-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb164-7"><a href="#cb164-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"STIM"</span>,</span>
<span id="cb164-8"><a href="#cb164-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb164-9"><a href="#cb164-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb164-10"><a href="#cb164-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb164-11"><a href="#cb164-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allVISIT_res.0.6_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb168-2"><a href="#cb168-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj,</span>
<span id="cb168-3"><a href="#cb168-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.adt"</span>,</span>
<span id="cb168-4"><a href="#cb168-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb168-5"><a href="#cb168-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb168-6"><a href="#cb168-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"group"</span>,</span>
<span id="cb168-7"><a href="#cb168-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"VISIT"</span>,</span>
<span id="cb168-8"><a href="#cb168-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb168-9"><a href="#cb168-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb168-10"><a href="#cb168-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb168-11"><a href="#cb168-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allConditions_res.0.6_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb172-2"><a href="#cb172-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj,</span>
<span id="cb172-3"><a href="#cb172-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.adt"</span>,</span>
<span id="cb172-4"><a href="#cb172-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb172-5"><a href="#cb172-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb172-6"><a href="#cb172-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"STIM"</span>),</span>
<span id="cb172-7"><a href="#cb172-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"VISIT"</span>,</span>
<span id="cb172-8"><a href="#cb172-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb172-9"><a href="#cb172-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb172-10"><a href="#cb172-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb172-11"><a href="#cb172-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values or values outside the scale range
(`geom_text_repel()`).</code></pre>
</div>
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_allConditionsAlternative_res.0.6_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj,</span>
<span id="cb176-3"><a href="#cb176-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.adt"</span>,</span>
<span id="cb176-4"><a href="#cb176-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb176-5"><a href="#cb176-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb176-6"><a href="#cb176-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"STIM"</span>),</span>
<span id="cb176-7"><a href="#cb176-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">split.by =</span> <span class="st">"group"</span>,</span>
<span id="cb176-8"><a href="#cb176-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">repel =</span> T,</span>
<span id="cb176-9"><a href="#cb176-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pt.size =</span> <span class="dv">1</span>,</span>
<span id="cb176-10"><a href="#cb176-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb176-11"><a href="#cb176-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">raster =</span> F)</span>
<span id="cb176-12"><a href="#cb176-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
<section id="determine-marker-expression-by-cluster-1" class="level3">
<h3 class="anchored" data-anchor-id="determine-marker-expression-by-cluster-1">Determine marker expression by cluster</h3>
<p>Find markers for every cell cluster compared to all remaining cells, report only the positive ones.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>obj.markers.adt <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(obj, <span class="at">only.pos =</span> <span class="cn">TRUE</span>, <span class="at">min.pct =</span> <span class="fl">0.25</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>)</span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a>obj.markers.adt <span class="sc">%&gt;%</span></span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb178-5"><a href="#cb178-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 158 × 7
# Groups:   cluster [16]
   p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene        
   &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;       
 1     0       1.69 0.993 0.764         0 0       CD4-RPA.T4  
 2     0       1.84 0.73  0.503         0 0       CD194       
 3     0       1.26 0.94  0.723         0 0       HuMsRt.CD278
 4     0       1.32 0.845 0.641         0 0       CD26        
 5     0       1.25 0.994 0.805         0 0       CD5         
 6     0       1.00 0.973 0.856         0 0       CD127       
 7     0       1.23 0.999 0.9           0 0       CD28        
 8     0       3.67 0.787 0.141         0 1       CD20-2H7    
 9     0       3.64 0.922 0.349         0 1       CD72        
10     0       3.18 0.965 0.394         0 1       CD268       
# ℹ 148 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(obj.markers.adt, <span class="at">file =</span> <span class="fu">file.path</span>(dir_save, <span class="st">"all_markers_snn0.6_adt.csv"</span>), <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="identify-markers-differentially-expressed-between-specific-clusters-of-interest-and-plot-differentially-expressed-features-as-relative-expression-and-as-counts.-1" class="level3">
<h3 class="anchored" data-anchor-id="identify-markers-differentially-expressed-between-specific-clusters-of-interest-and-plot-differentially-expressed-features-as-relative-expression-and-as-counts.-1">Identify markers differentially expressed between specific clusters of interest and plot differentially expressed features as relative expression and as counts.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list of all cluster identities in obj</span></span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a>cluster.ids.adt <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">Idents</span>(obj))</span>
<span id="cb181-3"><a href="#cb181-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-4"><a href="#cb181-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store cluster markers</span></span>
<span id="cb181-5"><a href="#cb181-5" aria-hidden="true" tabindex="-1"></a>cluster.markers.adt <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb181-6"><a href="#cb181-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-7"><a href="#cb181-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each Seurat cluster in obj</span></span>
<span id="cb181-8"><a href="#cb181-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-9"><a href="#cb181-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cluster.id <span class="cf">in</span> cluster.ids.adt) {</span>
<span id="cb181-10"><a href="#cb181-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb181-11"><a href="#cb181-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find markers for the current cluster</span></span>
<span id="cb181-12"><a href="#cb181-12" aria-hidden="true" tabindex="-1"></a>  cluster.markers.adt <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(obj, <span class="at">ident.1 =</span> cluster.id)</span>
<span id="cb181-13"><a href="#cb181-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb181-14"><a href="#cb181-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Assign a dynamic object name based on the cluster ID</span></span>
<span id="cb181-15"><a href="#cb181-15" aria-hidden="true" tabindex="-1"></a>  object.name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster.markers"</span>, cluster.id)</span>
<span id="cb181-16"><a href="#cb181-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb181-17"><a href="#cb181-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Export cluster markers to a CSV file</span></span>
<span id="cb181-18"><a href="#cb181-18" aria-hidden="true" tabindex="-1"></a>  csv_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(object.name, <span class="st">"_adt.csv"</span>)</span>
<span id="cb181-19"><a href="#cb181-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(cluster.markers.adt, <span class="at">file =</span> <span class="fu">file.path</span>(dir_save, <span class="fu">paste0</span>(csv_file)), <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb181-20"><a href="#cb181-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb181-21"><a href="#cb181-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot expression of top 10 differentially expressed genes for the current cluster</span></span>
<span id="cb181-22"><a href="#cb181-22" aria-hidden="true" tabindex="-1"></a>   vln_plot.adt <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb181-23"><a href="#cb181-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">features =</span> <span class="fu">rownames</span>(cluster.markers.adt)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb181-24"><a href="#cb181-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb181-25"><a href="#cb181-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb181-26"><a href="#cb181-26" aria-hidden="true" tabindex="-1"></a>  vln_plot_relexp.adt <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_VlnPlotRelExp_Top10_"</span>, object.name, <span class="st">"_adt.pdf"</span>)</span>
<span id="cb181-27"><a href="#cb181-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, vln_plot_relexp.adt), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb181-28"><a href="#cb181-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(vln_plot.adt)</span>
<span id="cb181-29"><a href="#cb181-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb181-30"><a href="#cb181-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb181-31"><a href="#cb181-31" aria-hidden="true" tabindex="-1"></a>  vln_plot.adt <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(obj,</span>
<span id="cb181-32"><a href="#cb181-32" aria-hidden="true" tabindex="-1"></a>          <span class="at">features =</span> <span class="fu">rownames</span>(cluster.markers.adt)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb181-33"><a href="#cb181-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">slot =</span> <span class="st">"counts"</span>, </span>
<span id="cb181-34"><a href="#cb181-34" aria-hidden="true" tabindex="-1"></a>          <span class="at">log =</span> <span class="cn">TRUE</span>, </span>
<span id="cb181-35"><a href="#cb181-35" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb181-36"><a href="#cb181-36" aria-hidden="true" tabindex="-1"></a>          <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb181-37"><a href="#cb181-37" aria-hidden="true" tabindex="-1"></a>  vln_plot_counts.adt <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_VlnPlotCounts_Top10_"</span>, object.name, <span class="st">"_adt.pdf"</span>)</span>
<span id="cb181-38"><a href="#cb181-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, vln_plot_counts.adt), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb181-39"><a href="#cb181-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(vln_plot.adt)</span>
<span id="cb181-40"><a href="#cb181-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb181-41"><a href="#cb181-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb181-42"><a href="#cb181-42" aria-hidden="true" tabindex="-1"></a>  feature_plot.adt <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(obj,</span>
<span id="cb181-43"><a href="#cb181-43" aria-hidden="true" tabindex="-1"></a>            <span class="at">features =</span> <span class="fu">rownames</span>(cluster.markers.adt)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], </span>
<span id="cb181-44"><a href="#cb181-44" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb181-45"><a href="#cb181-45" aria-hidden="true" tabindex="-1"></a>            <span class="at">label.size =</span> <span class="dv">4</span>,</span>
<span id="cb181-46"><a href="#cb181-46" aria-hidden="true" tabindex="-1"></a>            <span class="at">split.by =</span> <span class="st">"group"</span>,</span>
<span id="cb181-47"><a href="#cb181-47" aria-hidden="true" tabindex="-1"></a>            <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb181-48"><a href="#cb181-48" aria-hidden="true" tabindex="-1"></a>  feature_plots.adt <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"DEG_FeaturePlot_Top10_"</span>, object.name, <span class="st">"_adt.pdf"</span>)</span>
<span id="cb181-49"><a href="#cb181-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, feature_plots.adt), <span class="at">height =</span> <span class="dv">24</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb181-50"><a href="#cb181-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(feature_plot.adt)</span>
<span id="cb181-51"><a href="#cb181-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb181-52"><a href="#cb181-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb181-53"><a href="#cb181-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb181-54"><a href="#cb181-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-55"><a href="#cb181-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot a heatmap to look at the top 10 differentially expressed genes across all clusters</span></span>
<span id="cb181-56"><a href="#cb181-56" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"DEG_Heatmap_AllClusters_Top10_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">30</span>, <span class="at">width =</span> <span class="dv">20</span>)</span>
<span id="cb181-57"><a href="#cb181-57" aria-hidden="true" tabindex="-1"></a>obj.markers.adt <span class="sc">%&gt;%</span></span>
<span id="cb181-58"><a href="#cb181-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb181-59"><a href="#cb181-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">10</span>, <span class="at">wt =</span> avg_log2FC) <span class="ot">-&gt;</span> top10</span>
<span id="cb181-60"><a href="#cb181-60" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(obj, <span class="at">features =</span> top10<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb181-61"><a href="#cb181-61" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all markers distinguishing an interesting cluster from main PBMC clusters</span></span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a>cluster1vPBMC.markers.adt <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(obj, <span class="at">ident.1 =</span> <span class="dv">1</span>, <span class="at">ident.2 =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">11</span>))</span>
<span id="cb183-3"><a href="#cb183-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cluster1vPBMC.markers.adt, <span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         p_val avg_log2FC pct.1 pct.2 p_val_adj
CD20-2H7     0   3.663698 0.787 0.140         0
CD72         0   3.684997 0.922 0.344         0
CD268        0   3.178803 0.965 0.393         0
CD22         0   3.169480 0.972 0.409         0
CD21         0   2.925489 0.931 0.403         0
CD7          0  -2.486333 0.434 0.797         0
IgD          0   2.520340 0.872 0.516         0
CD40         0   1.941884 0.976 0.635         0
CD32         0   1.580264 0.951 0.629         0
CD185        0   1.262641 0.961 0.696         0</code></pre>
</div>
</div>
</section>
<section id="assign-cell-types-1" class="level3">
<h3 class="anchored" data-anchor-id="assign-cell-types-1">Assign cell types</h3>
<p>Predict cell types based on expression using sc-Type</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene set preparation function </span></span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R"</span>)</span>
<span id="cb185-3"><a href="#cb185-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-4"><a href="#cb185-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load cell type annotation function</span></span>
<span id="cb185-5"><a href="#cb185-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Prepare gene sets from input cell marker file. This uses the default of in-built cell marker DB.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb186"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb186-1"><a href="#cb186-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DB file</span></span>
<span id="cb186-2"><a href="#cb186-2" aria-hidden="true" tabindex="-1"></a><span class="co">#db_ &lt;- "https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx";</span></span>
<span id="cb186-3"><a href="#cb186-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-4"><a href="#cb186-4" aria-hidden="true" tabindex="-1"></a>db_ <span class="ot">&lt;-</span> <span class="st">"data/ScTypeDB_full_CYedit6.xlsx"</span></span>
<span id="cb186-5"><a href="#cb186-5" aria-hidden="true" tabindex="-1"></a>tissue <span class="ot">&lt;-</span> <span class="st">"Immune system"</span> <span class="co"># e.g. Immune system, Pancreas, Liver, Eye, Kidney, Brain, Lung, Adrenal, Heart, Intestine, Muscle, Placenta, Spleen, Stomach, Thymus </span></span>
<span id="cb186-6"><a href="#cb186-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-7"><a href="#cb186-7" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare gene sets</span></span>
<span id="cb186-8"><a href="#cb186-8" aria-hidden="true" tabindex="-1"></a>gs_list <span class="ot">&lt;-</span> <span class="fu">gene_sets_prepare</span>(db_, tissue)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkGeneSymbols(markers_all): Human gene symbols should be all
upper-case except for the 'orf' in open reading frames. The case of some
letters was corrected.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkGeneSymbols(markers_all): Human gene symbols should be all
upper-case except for the 'orf' in open reading frames. The case of some
letters was corrected.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkGeneSymbols(markers_all): Human gene symbols should be all
upper-case except for the 'orf' in open reading frames. The case of some
letters was corrected.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols
Warning in checkGeneSymbols(markers_all): x contains non-approved gene symbols</code></pre>
</div>
</div>
<p>Run sc-Type.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="do">### CHANGE obj@meta.data$seurat_clusters TO Idents(obj)</span></span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-3"><a href="#cb194-3" aria-hidden="true" tabindex="-1"></a><span class="co"># check Seurat object version (scRNA-seq matrix extracted differently in Seurat v4/v5)</span></span>
<span id="cb194-4"><a href="#cb194-4" aria-hidden="true" tabindex="-1"></a>seurat_package_v5 <span class="ot">&lt;-</span> <span class="fu">isFALSE</span>(<span class="st">'counts'</span> <span class="sc">%in%</span> <span class="fu">names</span>(<span class="fu">attributes</span>(obj[[<span class="st">"ADT"</span>]])));</span>
<span id="cb194-5"><a href="#cb194-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">sprintf</span>(<span class="st">"Seurat object %s is used"</span>, <span class="fu">ifelse</span>(seurat_package_v5, <span class="st">"v5"</span>, <span class="st">"v4"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Seurat object v5 is used"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb196"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb196-1"><a href="#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extract scaled scRNA-seq matrix</span></span>
<span id="cb196-2"><a href="#cb196-2" aria-hidden="true" tabindex="-1"></a>scRNAseqData_scaled.adt <span class="ot">&lt;-</span> <span class="cf">if</span> (seurat_package_v5) <span class="fu">as.matrix</span>(obj[[<span class="st">"ADT"</span>]]<span class="sc">$</span>scale.data) <span class="cf">else</span> <span class="fu">as.matrix</span>(obj[[<span class="st">"ADT"</span>]]<span class="sc">@</span>scale.data)</span>
<span id="cb196-3"><a href="#cb196-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-4"><a href="#cb196-4" aria-hidden="true" tabindex="-1"></a><span class="co"># run ScType</span></span>
<span id="cb196-5"><a href="#cb196-5" aria-hidden="true" tabindex="-1"></a>es.max.adt <span class="ot">&lt;-</span> <span class="fu">sctype_score</span>(<span class="at">scRNAseqData =</span> scRNAseqData_scaled.adt, <span class="at">scaled =</span> <span class="cn">TRUE</span>, <span class="at">gs =</span> gs_list<span class="sc">$</span>gs_positive, <span class="at">gs2 =</span> gs_list<span class="sc">$</span>gs_negative)</span>
<span id="cb196-6"><a href="#cb196-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-7"><a href="#cb196-7" aria-hidden="true" tabindex="-1"></a><span class="co"># merge by cluster</span></span>
<span id="cb196-8"><a href="#cb196-8" aria-hidden="true" tabindex="-1"></a>cL_resutls.adt <span class="ot">&lt;-</span> <span class="fu">do.call</span>(<span class="st">"rbind"</span>, <span class="fu">lapply</span>(<span class="fu">unique</span>(obj<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters), <span class="cf">function</span>(cl){</span>
<span id="cb196-9"><a href="#cb196-9" aria-hidden="true" tabindex="-1"></a>  es.max.cl.adt <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">rowSums</span>(es.max.adt[ ,<span class="fu">rownames</span>(obj<span class="sc">@</span>meta.data[obj<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters<span class="sc">==</span>cl, ])]), <span class="at">decreasing =</span> <span class="sc">!</span><span class="dv">0</span>)</span>
<span id="cb196-10"><a href="#cb196-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="fu">data.frame</span>(<span class="at">cluster =</span> cl, <span class="at">type =</span> <span class="fu">names</span>(es.max.cl.adt), <span class="at">scores =</span> es.max.cl.adt, <span class="at">ncells =</span> <span class="fu">sum</span>(obj<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters<span class="sc">==</span>cl)), <span class="dv">10</span>)</span>
<span id="cb196-11"><a href="#cb196-11" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb196-12"><a href="#cb196-12" aria-hidden="true" tabindex="-1"></a>sctype_scores.adt <span class="ot">&lt;-</span> cL_resutls.adt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">1</span>, <span class="at">wt =</span> scores)  </span>
<span id="cb196-13"><a href="#cb196-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-14"><a href="#cb196-14" aria-hidden="true" tabindex="-1"></a><span class="co"># set low-confident (low ScType score) clusters to "unknown"</span></span>
<span id="cb196-15"><a href="#cb196-15" aria-hidden="true" tabindex="-1"></a>sctype_scores.adt<span class="sc">$</span>type[<span class="fu">as.numeric</span>(<span class="fu">as.character</span>(sctype_scores.adt<span class="sc">$</span>scores)) <span class="sc">&lt;</span> sctype_scores.adt<span class="sc">$</span>ncells<span class="sc">/</span><span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">"Unknown"</span></span>
<span id="cb196-16"><a href="#cb196-16" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(sctype_scores.adt[,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 48 × 3
# Groups:   cluster [38]
   cluster type                        scores
   &lt;fct&gt;   &lt;chr&gt;                        &lt;dbl&gt;
 1 0       CD8a/a                       6747.
 2 0       Treg(diff)                   6747.
 3 8       ILC2                         3711.
 4 8       MAIT cells                   3711.
 5 35      Classical monocytes           197.
 6 35      Intermediate monocytes        197.
 7 23      Tem/Temra cytotoxic T cells   409.
 8 10      pDC                          5257.
 9 5       Cycling B cells              9049.
10 6       Unknown                       140.
# ℹ 38 more rows</code></pre>
</div>
</div>
<p>Overlay cell type IDs on UMAP.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a>obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_classification.adt <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> <span class="fu">unique</span>(sctype_scores.adt<span class="sc">$</span>cluster)){</span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a>  cl_type <span class="ot">=</span> sctype_scores.adt[sctype_scores.adt<span class="sc">$</span>cluster<span class="sc">==</span>j,]; </span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a>  obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_classification.adt[obj<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters <span class="sc">==</span> j] <span class="ot">=</span> <span class="fu">as.character</span>(cl_type<span class="sc">$</span>type[<span class="dv">1</span>])</span>
<span id="cb198-5"><a href="#cb198-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb198-6"><a href="#cb198-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-7"><a href="#cb198-7" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_annotated_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">7</span>, <span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb198-8"><a href="#cb198-8" aria-hidden="true" tabindex="-1"></a>umap_annotated_adt <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">group.by =</span> <span class="st">'sctype_classification.adt'</span>)</span>
<span id="cb198-9"><a href="#cb198-9" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(umap_annotated_adt)</span>
<span id="cb198-10"><a href="#cb198-10" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb198-11"><a href="#cb198-11" aria-hidden="true" tabindex="-1"></a>umap_annotated_adt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-69-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(dir_save, <span class="st">"umap_annotated_splitgroup_adt.pdf"</span>), <span class="at">height =</span> <span class="dv">7</span>, <span class="at">width =</span> <span class="dv">15</span>)</span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a>umap_annotated_adt_group <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">"umap.adt"</span>, <span class="at">split.by =</span> <span class="st">"group"</span>, <span class="at">group.by =</span> <span class="st">'sctype_classification.adt'</span>)</span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(umap_annotated_adt_group)</span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">dev.off</span>())</span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a>umap_annotated_adt_group</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-70-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="multimodal" class="level2">
<h2 class="anchored" data-anchor-id="multimodal">Multimodal</h2>
<section id="identify-multimodal-neighbors" class="level3">
<h3 class="anchored" data-anchor-id="identify-multimodal-neighbors">Identify multimodal neighbors</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindMultiModalNeighbors</span>(</span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a>  obj, <span class="at">reduction.list =</span> <span class="fu">list</span>(<span class="st">"pca"</span>, <span class="st">"pca.adt"</span>), </span>
<span id="cb200-3"><a href="#cb200-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">dims.list =</span> <span class="fu">list</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">18</span>), <span class="at">modality.weight.name =</span> <span class="st">"RNA.weight"</span></span>
<span id="cb200-4"><a href="#cb200-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cell-specific modality weights</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Finding 20 nearest neighbors for each modality.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating kernel bandwidths</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in FindMultiModalNeighbors(obj, reduction.list = list("pca",
"pca.adt"), : The number of provided modality.weight.name is not equal to the
number of modalities. RNA.weight ADT.weight are used to store the modality
weights</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Finding multimodal neighbors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Constructing multimodal KNN graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Constructing multimodal SNN graph</code></pre>
</div>
</div>
</section>
<section id="cluster-and-plot-based-on-rna-and-adt-assays" class="level3">
<h3 class="anchored" data-anchor-id="cluster-and-plot-based-on-rna-and-adt-assays">Cluster and plot based on RNA and ADT assays</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb208"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">nn.name =</span> <span class="st">"weighted.nn"</span>, <span class="at">reduction.name =</span> <span class="st">"wnn.umap"</span>, <span class="at">reduction.key =</span> <span class="st">"wnnUMAP_"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>00:34:22 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>00:34:25 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 20
00:34:27 Initializing from normalized Laplacian + noise (using RSpectra)
00:34:28 Commencing optimization for 200 epochs, with 2156654 positive edges
00:35:09 Optimization finished</code></pre>
</div>
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(obj, <span class="at">graph.name =</span> <span class="st">"wsnn"</span>, <span class="at">algorithm =</span> <span class="dv">3</span>, <span class="at">resolution =</span> <span class="dv">2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb211-2"><a href="#cb211-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-3"><a href="#cb211-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">'wnn.umap'</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"seurat_clusters"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-72-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">'wnn.umap'</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-73-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">'wnn.umap'</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"PID"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-73-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">'wnn.umap'</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-73-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">'wnn.umap'</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"VISIT_STIM"</span>, <span class="st">"seurat_clusters"</span>), <span class="at">split.by =</span> <span class="st">"group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="CITEseqHIVE_Seurat_ClusterIDcells_Multimodal_files/figure-html/unnamed-chunk-73-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="overlay-adt-expression-over-rna-clustered-umap." class="level3">
<h3 class="anchored" data-anchor-id="overlay-adt-expression-over-rna-clustered-umap.">Overlay ADT expression over RNA-clustered UMAP.</h3>
</section>
<section id="save-rdata-and-rds" class="level3">
<h3 class="anchored" data-anchor-id="save-rdata-and-rds">Save Rdata and rds</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(obj, <span class="at">file =</span> <span class="fu">paste0</span>(dir_save, <span class="st">"Seurat_ClusterIDCells_obj.Rdata"</span>))</span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(obj, <span class="at">file =</span> <span class="fu">paste0</span>(dir_save, <span class="st">"Seurat_ClusterIDcells_obj.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>